Last login: Thu Mar 23 10:21:18 on ttys004
rhamilto@ctr $ ssh ctr-bfx -l rsh46
Welcome to Ubuntu 16.04.2 LTS (GNU/Linux 4.4.0-64-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

23 packages can be updated.
0 updates are security updates.


*** System restart required ***
Last login: Thu Mar 23 10:49:15 2017 from 172.29.28.97
rsh46@ctr-bfx $ cd /storage/Genomes/
rsh46@ctr-bfx $ s
s: command not found
rsh46@ctr-bfx $ ls 
Drosophila_melanogaster  Escherichia_coli  Homo_sapiens  LambdaPhage  Mus_musculus  PhiX
rsh46@ctr-bfx $ cd Mus_musculus/
rsh46@ctr-bfx $ ls
CAST_Eij  GRCm37  GRCm38  GRCm38_ERCC
rsh46@ctr-bfx $ 
rsh46@ctr-bfx $ 
rsh46@ctr-bfx $ 
rsh46@ctr-bfx $ ls *
CAST_Eij:
Bisulfite_Genome                                       Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.15.fa
CAST_EiJ.1.bt2                                         Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.16.fa
CAST_EiJ.2.bt2                                         Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.17.fa
CAST_EiJ.3.bt2                                         Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.18.fa
CAST_EiJ.4.bt2                                         Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.19.fa
CAST_EiJ.rev.1.bt2                                     Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.1.fa
CAST_EiJ.rev.2.bt2                                     Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.2.fa
genomic_nucleotide_frequencies.txt                     Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.3.fa
Mus_musculus_casteij.CAST_EiJ_v1.86.gtf                Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.4.fa
Mus_musculus_casteij.CAST_EiJ_v1.cdna.all.fa.gz        Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.5.fa
Mus_musculus_casteij.CAST_EiJ_v1.cdna.all.idx          Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.6.fa
Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.10.fa  Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.7.fa
Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.11.fa  Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.8.fa
Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.12.fa  Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.9.fa
Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.13.fa  Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.all.fa
Mus_musculus_casteij.CAST_EiJ_v1.dna.chromosome.14.fa  Mus_musculus_casteij.CAST_EiJ_v1.dna.toplevel.fa.gz

GRCm37:
gencode.vM2.2wayconspseudos.gtf                  mm9_genes_ensembl_exons.bed
gencode.vM2.annotation.gtf                       mm9_genes_ensembl_exons.sorted.bed
gencode.vM2.chr_patch_hapl_scaff.annotation.gtf  mm9_genes_ensembl_intron.bed
gencode.vM2.long_noncoding_RNAs.gtf              mm9_genes_ensembl_intron.sorted.bed
mm9_genes_ensembl_3UTR.bed                       mm9_genes_ensembl.sorted.bed
mm9_genes_ensembl_3UTR.sorted.bed                mm9_genes_ensembl_UPSTREAM.bed
mm9_genes_ensembl_5UTR.bed                       mm9_genes_ensembl_UPSTREAM.sorted.bed
mm9_genes_ensembl_5UTR.sorted.bed                NCBI37_mm9.gtf
mm9_genes_ensembl.bed                            NCBI37_mm9.gtf.gz

GRCm38:
Bisulfite_Genome                                      Mus_musculus.GRCm38.dna.chromosome.17.fa
genomic_nucleotide_frequencies.txt                    Mus_musculus.GRCm38.dna.chromosome.18.fa
GRCm38.1.bt2                                          Mus_musculus.GRCm38.dna.chromosome.19.fa
GRCm38.2.bt2                                          Mus_musculus.GRCm38.dna.chromosome.1.fa
GRCm38.3.bt2                                          Mus_musculus.GRCm38.dna.chromosome.2.fa
GRCm38.4.bt2                                          Mus_musculus.GRCm38.dna.chromosome.3.fa
GRCm38.rev.1.bt2                                      Mus_musculus.GRCm38.dna.chromosome.4.fa
GRCm38.rev.2.bt2                                      Mus_musculus.GRCm38.dna.chromosome.5.fa
Mus_musculus.GRCm38.84.chr_patch_hapl_scaff_ERCC.gtf  Mus_musculus.GRCm38.dna.chromosome.6.fa
Mus_musculus.GRCm38.84.gtf                            Mus_musculus.GRCm38.dna.chromosome.7.fa
Mus_musculus.GRCm38.cdna.all.fa                       Mus_musculus.GRCm38.dna.chromosome.8.fa
Mus_musculus.GRCm38.cdna.all.idx                      Mus_musculus.GRCm38.dna.chromosome.9.fa
Mus_musculus.GRCm38.dna.chromosome.10.fa              Mus_musculus.GRCm38.dna.chromosome.all.dict
Mus_musculus.GRCm38.dna.chromosome.11.fa              Mus_musculus.GRCm38.dna.chromosome.all.fa
Mus_musculus.GRCm38.dna.chromosome.12.fa              Mus_musculus.GRCm38.dna.chromosome.all.fa.fai
Mus_musculus.GRCm38.dna.chromosome.13.fa              Mus_musculus.GRCm38.dna.chromosome.MT.fa
Mus_musculus.GRCm38.dna.chromosome.14.fa              Mus_musculus.GRCm38.dna.chromosome.X.fa
Mus_musculus.GRCm38.dna.chromosome.15.fa              Mus_musculus.GRCm38.dna.chromosome.Y.fa
Mus_musculus.GRCm38.dna.chromosome.16.fa

GRCm38_ERCC:
BT2IDX.stderr      GRCm38_ERCC.4.bt2   GRCm38_ERCC.rev.1.bt2  Mus_musculus.GRCm38.cdna.all_ERCC.fa.gz
BT2IDX.stdout      GRCm38_ERCC.4.ht2   GRCm38_ERCC.rev.2.bt2  Mus_musculus.GRCm38.cdna.all_ERCC.idx
bt2_indexer.sh     GRCm38_ERCC.5.ht2   HS2IDX.stderr          Mus_musculus.GRCm38.cdna.all.fa
GRCm38_ERCC.1.bt2  GRCm38_ERCC.6.ht2   HS2IDX.stdout          STAR
GRCm38_ERCC.1.ht2  GRCm38_ERCC.7.ht2   hs2_indexer.sh         STARIDX.stderr
GRCm38_ERCC.2.bt2  GRCm38_ERCC.8.ht2   KALLIDX.stderr         STARIDX.stdout
GRCm38_ERCC.2.ht2  GRCm38_ERCC.fa      KALLIDX.stdout         star_indexer.sh
GRCm38_ERCC.3.bt2  GRCm38_ERCC.fa.fai  kall_indexer.sh
GRCm38_ERCC.3.ht2  GRCm38_ERCC.gtf     Log.out
rsh46@ctr-bfx $ cd GRCm38
rsh46@ctr-bfx $ ls
Bisulfite_Genome                                      Mus_musculus.GRCm38.dna.chromosome.17.fa
genomic_nucleotide_frequencies.txt                    Mus_musculus.GRCm38.dna.chromosome.18.fa
GRCm38.1.bt2                                          Mus_musculus.GRCm38.dna.chromosome.19.fa
GRCm38.2.bt2                                          Mus_musculus.GRCm38.dna.chromosome.1.fa
GRCm38.3.bt2                                          Mus_musculus.GRCm38.dna.chromosome.2.fa
GRCm38.4.bt2                                          Mus_musculus.GRCm38.dna.chromosome.3.fa
GRCm38.rev.1.bt2                                      Mus_musculus.GRCm38.dna.chromosome.4.fa
GRCm38.rev.2.bt2                                      Mus_musculus.GRCm38.dna.chromosome.5.fa
Mus_musculus.GRCm38.84.chr_patch_hapl_scaff_ERCC.gtf  Mus_musculus.GRCm38.dna.chromosome.6.fa
Mus_musculus.GRCm38.84.gtf                            Mus_musculus.GRCm38.dna.chromosome.7.fa
Mus_musculus.GRCm38.cdna.all.fa                       Mus_musculus.GRCm38.dna.chromosome.8.fa
Mus_musculus.GRCm38.cdna.all.idx                      Mus_musculus.GRCm38.dna.chromosome.9.fa
Mus_musculus.GRCm38.dna.chromosome.10.fa              Mus_musculus.GRCm38.dna.chromosome.all.dict
Mus_musculus.GRCm38.dna.chromosome.11.fa              Mus_musculus.GRCm38.dna.chromosome.all.fa
Mus_musculus.GRCm38.dna.chromosome.12.fa              Mus_musculus.GRCm38.dna.chromosome.all.fa.fai
Mus_musculus.GRCm38.dna.chromosome.13.fa              Mus_musculus.GRCm38.dna.chromosome.MT.fa
Mus_musculus.GRCm38.dna.chromosome.14.fa              Mus_musculus.GRCm38.dna.chromosome.X.fa
Mus_musculus.GRCm38.dna.chromosome.15.fa              Mus_musculus.GRCm38.dna.chromosome.Y.fa
Mus_musculus.GRCm38.dna.chromosome.16.fa
rsh46@ctr-bfx $ 
rsh46@ctr-bfx $ 
rsh46@ctr-bfx $ 
rsh46@ctr-bfx $ exit
logout
Connection to ctr-bfx closed.
rhamilto@ctr $ darwin

        <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
        <>                                                        <>
        <>                        !WARNING!                       <>
        <>                                                        <>
        <>                  HPCs Darwin Facility                  <>
        <>             Unauthorised Access Prohibited             <>
        <>     Use of this system constitutes acceptance of our   <>
        <>         policies - see http://www.hpc.cam.ac.uk/.      <>
        <>                                                        <>
        <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

Last login: Thu Mar 23 11:15:07 2017 from ctr-rsh46.pdn.private.cam.ac.uk
[rsh46@login-sand3 ~]$ cd ~/scratch/Programs/
[rsh46@login-sand3 Programs]$ cd hisat2-
-bash: cd: hisat2-: No such file or directory
[rsh46@login-sand3 Programs]$ ls
bamtools                      erne-2.1.1-linux         kallisto_linux-v0.43.0       Salmon-0.8.2_linux_x86_64
bbmap                         erne-2.1.1-linux.tar.gz  kallisto_linux-v0.43.1       samtools-1.3.1
bcftools-1.3.1                FastQC                   local                        seqtk
Bismark                       fastq_screen_v0.5.2      manta-0.29.6.centos5_x86_64  snpEff
bowtie2-2.2.9                 gmap-2017-03-17          MultiQC                      STAR
clusterflow                   hisat2-2.0.4             picard-tools-2.2.4           subread-1.5.0-p2-Linux-x86_64
cramtools-3.0                 hisat2-cba6e8cb          preseq_v2.0                  svtools
cufflinks-2.2.1.Linux_x86_64  HTSeq-0.6.1p1            pyBigWig                     telseq
deepTools                     htslib-1.3.1             pysam                        tophat-2.1.1.Linux_x86_64
Drop-seq_tools-1.12           infernal-1.1.2           qualimap_v2.2                trim_galore_zip
dupRadar                      kallisto_linux-v0.42.5   rrna_mt_count
[rsh46@login-sand3 Programs]$ rm erne-2.1.1-linux.tar.gz 
[rsh46@login-sand3 Programs]$ ls
bamtools                      erne-2.1.1-linux        kallisto_linux-v0.43.1       samtools-1.3.1
bbmap                         FastQC                  local                        seqtk
bcftools-1.3.1                fastq_screen_v0.5.2     manta-0.29.6.centos5_x86_64  snpEff
Bismark                       gmap-2017-03-17         MultiQC                      STAR
bowtie2-2.2.9                 hisat2-2.0.4            picard-tools-2.2.4           subread-1.5.0-p2-Linux-x86_64
clusterflow                   hisat2-cba6e8cb         preseq_v2.0                  svtools
cramtools-3.0                 HTSeq-0.6.1p1           pyBigWig                     telseq
cufflinks-2.2.1.Linux_x86_64  htslib-1.3.1            pysam                        tophat-2.1.1.Linux_x86_64
deepTools                     infernal-1.1.2          qualimap_v2.2                trim_galore_zip
Drop-seq_tools-1.12           kallisto_linux-v0.42.5  rrna_mt_count
dupRadar                      kallisto_linux-v0.43.0  Salmon-0.8.2_linux_x86_64
[rsh46@login-sand3 Programs]$ 
[rsh46@login-sand3 Programs]$ 
[rsh46@login-sand3 Programs]$ 
[rsh46@login-sand3 Programs]$ ls hisat2-2.0.4/
AUTHORS                  hisat2-build                            hisat2_extract_splice_sites.py  LICENSE
doc                      hisat2_build_genotype_genome.py         hisat2_genotype.py              MANUAL
example                  hisat2-build-l                          hisat2-inspect                  MANUAL.markdown
extract_exons.py         hisat2-build-l-debug                    hisat2-inspect-l                NEWS
extract_splice_sites.py  hisat2-build-s                          hisat2-inspect-l-debug          scripts
hisat2                   hisat2-build-s-debug                    hisat2-inspect-s                TUTORIAL
hisat2-align-l           hisat2_extract_exons.py                 hisat2-inspect-s-debug          VERSION
hisat2-align-l-debug     hisat2_extract_HLA_vars.py              hisat2_simulate_reads.py
hisat2-align-s           hisat2_extract_snps_haplotypes_UCSC.py  hisat2_test_BRCA_genotyping.py
hisat2-align-s-debug     hisat2_extract_snps_haplotypes_VCF.py   hisat2_test_HLA_genotyping.py
[rsh46@login-sand3 Programs]$ ls -hl
total 192K
drwxr-xr-x  9 rsh46 rsh46 4.0K Jun  7  2016 bamtools
drwxr-xr-x  7 rsh46 rsh46  12K Oct 14 11:08 bbmap
drwxr-xr-x  6 rsh46 rsh46 4.0K Jun  8  2016 bcftools-1.3.1
drwxr-xr-x  4 rsh46 rsh46 4.0K Jul 28  2016 Bismark
drwxrwxr-x  5 rsh46 rsh46 4.0K Jun  8  2016 bowtie2-2.2.9
drwxr-xr-x  8 rsh46 rsh46 4.0K Sep  7  2016 clusterflow
drwxrwxr-x  6 rsh46 rsh46 4.0K Aug  4  2015 cramtools-3.0
drwxrwxr-x  2 rsh46 rsh46 4.0K Apr 30  2014 cufflinks-2.2.1.Linux_x86_64
drwxr-xr-x 13 rsh46 rsh46 4.0K Jun  6  2016 deepTools
drwxr-xr-x  4 rsh46 rsh46 4.0K Dec 14 09:58 Drop-seq_tools-1.12
drwxrwxr-x  2 rsh46 rsh46 4.0K Nov 21 12:34 dupRadar
drwxrwxr-x  4 rsh46 rsh46 4.0K May  3  2016 erne-2.1.1-linux
drwxrwxr-x  8 rsh46 rsh46 4.0K May  9  2016 FastQC
drwxr-xr-x  2 rsh46 rsh46 4.0K Mar  1 14:36 fastq_screen_v0.5.2
drwxr-xr-x  7 rsh46 rsh46 4.0K Mar 22 12:41 gmap-2017-03-17
drwxrwxr-x  5 rsh46 rsh46 4.0K May 18  2016 hisat2-2.0.4
drwxr-xr-x 12 rsh46 rsh46  12K Mar 22 11:55 hisat2-cba6e8cb
drwxrwxr-x  8 rsh46 rsh46 4.0K Jun  8  2016 HTSeq-0.6.1p1
drwxr-xr-x  5 rsh46 rsh46 4.0K Jun  6  2016 htslib-1.3.1
drwxr-xr-x 11 rsh46 rsh46 4.0K Aug 24  2016 infernal-1.1.2
drwxrwxr-x  3 rsh46 rsh46 4.0K Apr  4  2016 kallisto_linux-v0.42.5
drwxrwxr-x  3 rsh46 rsh46 4.0K Jun  2  2016 kallisto_linux-v0.43.0
drwxr-xr-x  3 rsh46 rsh46 4.0K Mar 22 11:41 kallisto_linux-v0.43.1
drwxrwxrwx  5 rsh46 rsh46 4.0K May 10  2016 local
drwxr-xr-x  6 rsh46 rsh46 4.0K Apr 13  2016 manta-0.29.6.centos5_x86_64
drwxr-xr-x  8 rsh46 rsh46 4.0K Sep 27 10:54 MultiQC
drwxr-xr-x  2 rsh46 rsh46 4.0K Apr 27  2016 picard-tools-2.2.4
drwxr-xr-x  2 rsh46 rsh46 4.0K Oct 29  2015 preseq_v2.0
drwxr-xr-x  8 rsh46 rsh46 4.0K May  9  2016 pyBigWig
drwxr-xr-x 16 rsh46 rsh46 4.0K Jun  6  2016 pysam
drwxrwxr-x  5 rsh46 rsh46 4.0K Nov 18 12:16 qualimap_v2.2
drwxrwxr-x  3 rsh46 rsh46 4.0K Feb 27 12:46 rrna_mt_count
drwxr-xr-x  4 rsh46 rsh46 4.0K Mar 22 11:41 Salmon-0.8.2_linux_x86_64
drwxr-xr-x  7 rsh46 rsh46  12K May  9  2016 samtools-1.3.1
drwxr-xr-x  3 rsh46 rsh46 4.0K Jun 16  2016 seqtk
drwxr-xr-x  5 rsh46 rsh46 4.0K May 25  2016 snpEff
drwxr-xr-x  7 rsh46 rsh46 4.0K May  6  2016 STAR
drwxr-xr-x  6 rsh46 rsh46 4.0K Apr 13  2016 subread-1.5.0-p2-Linux-x86_64
drwxrwxr-x  3 rsh46 rsh46 4.0K Jun  1  2016 svtools
drwxr-xr-x  5 rsh46 rsh46 4.0K Jun  7  2016 telseq
drwxrwxr-x  4 rsh46 rsh46 4.0K Apr  2  2016 tophat-2.1.1.Linux_x86_64
drwxrwxr-x  2 rsh46 rsh46 4.0K May  9  2016 trim_galore_zip
[rsh46@login-sand3 Programs]$ rm -r hisat2-cba6e8cb
[rsh46@login-sand3 Programs]$ ls
bamtools                      erne-2.1.1-linux        local                        seqtk
bbmap                         FastQC                  manta-0.29.6.centos5_x86_64  snpEff
bcftools-1.3.1                fastq_screen_v0.5.2     MultiQC                      STAR
Bismark                       gmap-2017-03-17         picard-tools-2.2.4           subread-1.5.0-p2-Linux-x86_64
bowtie2-2.2.9                 hisat2-2.0.4            preseq_v2.0                  svtools
clusterflow                   HTSeq-0.6.1p1           pyBigWig                     telseq
cramtools-3.0                 htslib-1.3.1            pysam                        tophat-2.1.1.Linux_x86_64
cufflinks-2.2.1.Linux_x86_64  infernal-1.1.2          qualimap_v2.2                trim_galore_zip
deepTools                     kallisto_linux-v0.42.5  rrna_mt_count
Drop-seq_tools-1.12           kallisto_linux-v0.43.0  Salmon-0.8.2_linux_x86_64
dupRadar                      kallisto_linux-v0.43.1  samtools-1.3.1
[rsh46@login-sand3 Programs]$ cd hisat2-2.0.4/
[rsh46@login-sand3 hisat2-2.0.4]$ ls
AUTHORS                  hisat2-build                            hisat2_extract_splice_sites.py  LICENSE
doc                      hisat2_build_genotype_genome.py         hisat2_genotype.py              MANUAL
example                  hisat2-build-l                          hisat2-inspect                  MANUAL.markdown
extract_exons.py         hisat2-build-l-debug                    hisat2-inspect-l                NEWS
extract_splice_sites.py  hisat2-build-s                          hisat2-inspect-l-debug          scripts
hisat2                   hisat2-build-s-debug                    hisat2-inspect-s                TUTORIAL
hisat2-align-l           hisat2_extract_exons.py                 hisat2-inspect-s-debug          VERSION
hisat2-align-l-debug     hisat2_extract_HLA_vars.py              hisat2_simulate_reads.py
hisat2-align-s           hisat2_extract_snps_haplotypes_UCSC.py  hisat2_test_BRCA_genotyping.py
hisat2-align-s-debug     hisat2_extract_snps_haplotypes_VCF.py   hisat2_test_HLA_genotyping.py
[rsh46@login-sand3 hisat2-2.0.4]$ 
[rsh46@login-sand3 hisat2-2.0.4]$ 
[rsh46@login-sand3 hisat2-2.0.4]$ 
[rsh46@login-sand3 hisat2-2.0.4]$ vi ~/.bashrc 
[rsh46@login-sand3 hisat2-2.0.4]$ source ~/.bashrc 
[rsh46@login-sand3 hisat2-2.0.4]$ hisat2
No index, query, or output file specified!
HISAT2 version 2.0.4 by Daehwan Kim (infphilo@gmail.com, www.ccb.jhu.edu/people/infphilo)
Usage: 
  hisat2 [options]* -x <ht2-idx> {-1 <m1> -2 <m2> | -U <r> | --sra-acc <SRA accession number>} [-S <sam>]

  <ht2-idx>  Index filename prefix (minus trailing .X.ht2).
  <m1>       Files with #1 mates, paired with files in <m2>.
             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).
  <m2>       Files with #2 mates, paired with files in <m1>.
             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).
  <r>        Files with unpaired reads.
             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).
  <SRA accession number>        Comma-separated list of SRA accession numbers, e.g. --sra-acc SRR353653,SRR353654.
  <sam>      File for SAM output (default: stdout)

  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be
  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.

Options (defaults in parentheses):

 Input:
  -q                 query input files are FASTQ .fq/.fastq (default)
  --qseq             query input files are in Illumina's qseq format
  -f                 query input files are (multi-)FASTA .fa/.mfa
  -r                 query input files are raw one-sequence-per-line
  -c                 <m1>, <m2>, <r> are sequences themselves, not files
  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)
  -u/--upto <int>    stop after first <int> reads/pairs (no limit)
  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)
  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)
  --phred33          qualities are Phred+33 (default)
  --phred64          qualities are Phred+64
  --int-quals        qualities encoded as space-delimited integers
  --sra-acc          SRA accession ID

 Alignment:
  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)
  --ignore-quals     treat all quality values as 30 on Phred scale (off)
  --nofw             do not align forward (original) version of read (off)
  --norc             do not align reverse-complement version of read (off)

 Spliced Alignment:
  --pen-cansplice <int>              penalty for a canonical splice site (0)
  --pen-noncansplice <int>           penalty for a non-canonical splice site (12)
  --pen-canintronlen <func>          penalty for long introns (G,-8,1) with canonical splice sites
  --pen-noncanintronlen <func>       penalty for long introns (G,-8,1) with noncanonical splice sites
  --min-intronlen <int>              minimum intron length (20)
  --max-intronlen <int>              maximum intron length (500000)
  --known-splicesite-infile <path>   provide a list of known splice sites
  --novel-splicesite-outfile <path>  report a list of splice sites
  --novel-splicesite-infile <path>   provide a list of novel splice sites
  --no-temp-splicesite               disable the use of splice sites found
  --no-spliced-alignment             disable spliced alignment
  --rna-strandness <string>          Specify strand-specific information (unstranded)
  --tmo                              Reports only those alignments within known transcriptome
  --dta                              Reports alignments tailored for transcript assemblers
  --dta-cufflinks                    Reports alignments tailored specifically for cufflinks

 Scoring:
  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) 
  --mp <int>,<int>   max and min penalties for mismatch; lower qual = lower penalty <2,6>
  --sp <int>,<int>   max and min penalties for soft-clipping; lower qual = lower penalty <1,2>
  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)
  --rdg <int>,<int>  read gap open, extend penalties (5,3)
  --rfg <int>,<int>  reference gap open, extend penalties (5,3)
  --score-min <func> min acceptable alignment score w/r/t read length
                     (L,0.0,-0.2)

 Reporting:
  (default)          look for multiple alignments, report best, with MAPQ
   OR
  -k <int>           report up to <int> alns per read; MAPQ not meaningful
   OR
  -a/--all           report all alignments; very slow, MAPQ not meaningful

 Paired-end:
  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)
  --no-mixed         suppress unpaired alignments for paired reads
  --no-discordant    suppress discordant alignments for paired reads

 Output:
  -t/--time          print wall-clock time taken by search phases
  --un <path>           write unpaired reads that didn't align to <path>
  --al <path>           write unpaired reads that aligned at least once to <path>
  --un-conc <path>      write pairs that didn't align concordantly to <path>
  --al-conc <path>      write pairs that aligned concordantly at least once to <path>
  (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.
  --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)
  --quiet            print nothing to stderr except serious errors
  --met-file <path>  send metrics to file at <path> (off)
  --met-stderr       send metrics to stderr (off)
  --met <int>        report internal counters & metrics every <int> secs (1)
  --no-head          supppress header lines, i.e. lines starting with @
  --no-sq            supppress @SQ header lines
  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field
  --rg <text>        add <text> ("lab:value") to @RG line of SAM header.
                     Note: @RG line only printed when --rg-id is set.
  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.

 Performance:
  -o/--offrate <int> override offrate of index; must be >= index's offrate
  -p/--threads <int> number of alignment threads to launch (1)
  --reorder          force SAM output order to match order of input reads
  --mm               use memory-mapped I/O for index; many 'bowtie's can share

 Other:
  --qc-filter        filter out reads that are bad according to QSEQ filter
  --seed <int>       seed for random number generator (0)
  --non-deterministic seed rand. gen. arbitrarily instead of using read attributes
  --remove-chrname   remove 'chr' from reference names in alignment
  --add-chrname      add 'chr' to reference names in alignment 
  --version          print version information and quit
  -h/--help          print this usage message
(ERR): hisat2-align exited with value 1
[rsh46@login-sand3 hisat2-2.0.4]$ 
[rsh46@login-sand3 hisat2-2.0.4]$ 
[rsh46@login-sand3 hisat2-2.0.4]$ erne2
-bash: erne2: command not found
[rsh46@login-sand3 hisat2-2.0.4]$ cd ../
[rsh46@login-sand3 Programs]$ ls
bamtools                      erne-2.1.1-linux        local                        seqtk
bbmap                         FastQC                  manta-0.29.6.centos5_x86_64  snpEff
bcftools-1.3.1                fastq_screen_v0.5.2     MultiQC                      STAR
Bismark                       gmap-2017-03-17         picard-tools-2.2.4           subread-1.5.0-p2-Linux-x86_64
bowtie2-2.2.9                 hisat2-2.0.4            preseq_v2.0                  svtools
clusterflow                   HTSeq-0.6.1p1           pyBigWig                     telseq
cramtools-3.0                 htslib-1.3.1            pysam                        tophat-2.1.1.Linux_x86_64
cufflinks-2.2.1.Linux_x86_64  infernal-1.1.2          qualimap_v2.2                trim_galore_zip
deepTools                     kallisto_linux-v0.42.5  rrna_mt_count
Drop-seq_tools-1.12           kallisto_linux-v0.43.0  Salmon-0.8.2_linux_x86_64
dupRadar                      kallisto_linux-v0.43.1  samtools-1.3.1
[rsh46@login-sand3 Programs]$ cd gmap-2017-03-17/
[rsh46@login-sand3 gmap-2017-03-17]$ ls
acinclude.m4  bin        config.log     configure     INSTALL   Makefile.am  NOTICE  tests  VERSION
aclocal.m4    ChangeLog  config.site    configure.ac  libtool   Makefile.in  README  TODO
AUTHORS       config     config.status  COPYING       Makefile  NEWS         src     util
[rsh46@login-sand3 gmap-2017-03-17]$ cd ../erne-2.1.1-linux/
[rsh46@login-sand3 erne-2.1.1-linux]$ ls
bin  share
[rsh46@login-sand3 erne-2.1.1-linux]$ cd bin/
[rsh46@login-sand3 bin]$ ls
erne-bs5  erne-create  erne-filter  erne-map  erne-meth
[rsh46@login-sand3 bin]$ cd ../
[rsh46@login-sand3 erne-2.1.1-linux]$ ls
bin  share
[rsh46@login-sand3 erne-2.1.1-linux]$ cd ../
[rsh46@login-sand3 Programs]$ ls
bamtools                      erne-2.1.1-linux        local                        seqtk
bbmap                         FastQC                  manta-0.29.6.centos5_x86_64  snpEff
bcftools-1.3.1                fastq_screen_v0.5.2     MultiQC                      STAR
Bismark                       gmap-2017-03-17         picard-tools-2.2.4           subread-1.5.0-p2-Linux-x86_64
bowtie2-2.2.9                 hisat2-2.0.4            preseq_v2.0                  svtools
clusterflow                   HTSeq-0.6.1p1           pyBigWig                     telseq
cramtools-3.0                 htslib-1.3.1            pysam                        tophat-2.1.1.Linux_x86_64
cufflinks-2.2.1.Linux_x86_64  infernal-1.1.2          qualimap_v2.2                trim_galore_zip
deepTools                     kallisto_linux-v0.42.5  rrna_mt_count
Drop-seq_tools-1.12           kallisto_linux-v0.43.0  Salmon-0.8.2_linux_x86_64
dupRadar                      kallisto_linux-v0.43.1  samtools-1.3.1
[rsh46@login-sand3 Programs]$ cd subread-1.5.0-p2-Linux-x86_64/
[rsh46@login-sand3 subread-1.5.0-p2-Linux-x86_64]$ ls
annotation  bin  doc  LICENSE  README.txt  test
[rsh46@login-sand3 subread-1.5.0-p2-Linux-x86_64]$ cd bin/
[rsh46@login-sand3 bin]$ ls
exactSNP  featureCounts  subindel  subjunc  subread-align  subread-buildindex  utilities
[rsh46@login-sand3 bin]$ erne-create
Both '--output-prefix', and '--fasta' are required!
Try "--help" for help
[rsh46@login-sand3 bin]$ erne-create --help
ERNE version 2.1.1

This is ERNE-CREATE.Usage: erne-create [--methyl-hash] [--former-hash] --fasta in.fa --output-prefix out [--k k] [--o o] [--bl bl]
Allowed options: :
  --help                produce help message
  --version             print version and exit
  --fasta arg           input file (can be repeated several time) [REQUIRED]
  --output-prefix arg   output reference file name in ERNE format. Suffix added
                        automatically: .ebh for dB-Hash reference, .ebm for 
                        dB-Hash methylated reference. [REQUIRED]
                        
  --methyl-hash         create reference for methylation search
  --k arg               [DEPRECATED] fingerprint length (default is 
                        autodetected: log4(bl*n/2), where n=text length);  the 
                        range of admissible values is 5-32. With --methyl-hash 
                        the range of admissible values is 10-64
  --bl arg              seed length (default: 20 for DNA search, 30 for 
                        BS-search)
  --o arg               offrate for SA pointers (default: 16).

[rsh46@login-sand3 bin]$ gmap_build
-k flag not specified, so building with default 15-mers

gmap_build: Builds a gmap database for a genome to be used by GMAP or GSNAP.
Part of GMAP package, version 2017-03-17.

A simplified alternative to using the program gmap_setup, which creates a Makefile.

Usage: gmap_build [options...] -d <genomename> <fasta_files>

Options:
    -D, --dir=STRING          Destination directory for installation (defaults to gmapdb directory specified at configure time)
    -d, --db=STRING           Genome name

    -n, --names=STRING        Substitute names for chromosomes, provided in a file.  The file should have one line
                                for each chromosome name to be changed, with the original FASTA name in column 1 and
                                the desired chromosome name in column 2.  This provides an easy way to change the
                                names of chromosomes, for example, to add or remove the "chr" prefix.  Column 2 may
                                be blank, which indicates no name change.  This file can also be combined with
                                --sort=names to provide a particular order for the chromosomes in the genome index.

    -M, --mdflag=STRING       Use MD file from NCBI for mapping contigs to chromosomal coordinates
    -C, --contigs-are-mapped  Find a chromosomal region in each FASTA header line.  Useful for contigs that have been mapped
                                to chromosomal coordinates.  Ignored if the --mdflag is provided.

    -z, --compression=STRING  Use given compression types (separated by commas; default is bitpack64)
                                bitpack64 - optimized for modern computers with SIMD instructions (recommended)
                                all - create all available compression types, currently bitpack64
                                none - do not compress offset files (I believe this is no longer supported)

    -k, --kmer=INT            k-mer value for genomic index (allowed: 15 or less, default is 15)
    -q INT                    sampling interval for genomoe (allowed: 1-3, default 3)

    -s, --sort=STRING         Sort chromosomes using given method:
                                none - use chromosomes as found in FASTA file(s)
                                alpha - sort chromosomes alphabetically (chr10 before chr 1)
                                numeric-alpha - chr1, chr1U, chr2, chrM, chrU, chrX, chrY
                                chrom - chr1, chr2, chrM, chrX, chrY, chr1U, chrU
                                names - sort chromosomes based on file provided to --names flag

    -g, --gunzip              Files are gzipped, so need to gunzip each file first
    -E, --fasta-pipe=STRING   Interpret argument as a command, instead of a list of FASTA files
    -w INT                    Wait (sleep) this many seconds after each step (default 2)

    -c, --circular=STRING     Circular chromosomes (either a list of chromosomes separated by a comma, or
                                a filename containing circular chromosomes, one per line).  If you use the
                                --names feature, then you should use the original name of the chromosome,
                                not the substitute name, for this option.                                                        

    -2, --altscaffold=STRING  File with alt scaffold info, listing alternate scaffolds, one per line, tab-delimited,
                                with the following fields: (1) alt_scaf_acc, (2) parent_name, (3) orientation,
                                (4) alt_scaf_start, (5) alt_scaf_stop, (6) parent_start, (7) parent_end.

    -e, --nmessages=INT       Maximum number of messages (warnings, contig reports) to report (default 50)

    --build-sarray=INT        Whether to build suffix array: 0=no, 1=yes (default)

Obsolete options:
    -T STRING                 Temporary build directory (may need to specify if you run out of space in your current directory)
                                This is no longer necessary, since gmap_build now builds directly in the destination
                                (or -D) directory.

Must specify genome database name with -d flag. at /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gmap_build line 71.
[rsh46@login-sand3 bin]$ subread-buildindex

Version 1.5.0-p2

Usage:

 ./subread-buildindex [options] -o <basename> {FASTA file1} [FASTA file2] ...

Required arguments:

    -o <basename>   base name of the index to be created

Optional arguments:

    -F              build a full index for the reference genome. 16bp subreads
                    will be extracted from every position of the reference
                    genome. Size of the index is typically 3 times the size of
                    index built from using the default setting.

    -B              create one block of index. The built index will not be split
                    into multiple pieces. This makes the largest amount of
                    memory be requested when running alignments, but it enables
                    the maximum mapping speed to be achieved. This option
                    overrides -M when it is provided as well.

    -M <int>        size of requested memory(RAM) in megabytes, 8000 by default.

    -f <int>        specify the threshold for removing uninformative subreads
                    (highly repetitive 16mers in the reference). 100 by default.

    -c              build a color-space index.

    -v              output version of the program.

For more information about these arguments, please refer to the User Manual.

[rsh46@login-sand3 bin]$ cd ~/scratch/Genomes/Homo_sapiens/GRCh38.82/
[rsh46@login-sand3 GRCh38.82]$ ls
GRCh38.82.1.bt2                              Homo_sapiens.GRCh38.dna.chromosome.20.fa.gz
GRCh38.82.2.bt2                              Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz
GRCh38.82.3.bt2                              Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz
GRCh38.82.4.bt2                              Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz
GRCh38.82.rev.1.bt2                          Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz
GRCh38.82.rev.2.bt2                          Homo_sapiens.GRCh38.dna.chromosome.4.fa.gz
Homo_sapiens.GRCh38.82.gtf                   Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz
Homo_sapiens.GRCh38.cdna.all.fa.gz           Homo_sapiens.GRCh38.dna.chromosome.6.fa.gz
Homo_sapiens.GRCh38.dna.chromosome.10.fa.gz  Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz
Homo_sapiens.GRCh38.dna.chromosome.11.fa.gz  Homo_sapiens.GRCh38.dna.chromosome.8.fa.gz
Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz  Homo_sapiens.GRCh38.dna.chromosome.9.fa.gz
Homo_sapiens.GRCh38.dna.chromosome.13.fa.gz  Homo_sapiens.GRCh38.dna.chromosome.all.fa
Homo_sapiens.GRCh38.dna.chromosome.14.fa.gz  Homo_sapiens.GRCh38.dna.chromosome.MT.fa.gz
Homo_sapiens.GRCh38.dna.chromosome.15.fa.gz  Homo_sapiens.GRCh38.dna.chromosome.X.fa.gz
Homo_sapiens.GRCh38.dna.chromosome.16.fa.gz  Homo_sapiens.GRCh38.dna.chromosome.Y.fa.gz
Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz  IDX.stderr
Homo_sapiens.GRCh38.dna.chromosome.18.fa.gz  IDX.stdout
Homo_sapiens.GRCh38.dna.chromosome.19.fa.gz  indexer.sh
Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz
[rsh46@login-sand3 GRCh38.82]$ myquota
Home    Usage: 9.35944% GB=3.74377
Scratch Usage: 85.0446% GB=870.857
[rsh46@login-sand3 GRCh38.82]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601722   IDX        rsh46         Running 6       23:31:12  Thu Mar 23 11:30:40

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================
5601770   IDX2       rsh46         Waiting 6       24:00:00  Thu Mar 23 11:59:04

Total Jobs: 2     Active Jobs: 1     Idle Jobs: 1     Blocked Jobs: 0   
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ ls -hltrog
total 7.5G
-rw-rw-r-- 1 1.4G Mar 23 11:16 Homo_sapiens.GRCh38.82.gtf
-rw-rw-r-- 1  38M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.11.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.10.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.18.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz
-rw-rw-r-- 1  24M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.16.fa.gz
-rw-rw-r-- 1  25M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.15.fa.gz
-rw-rw-r-- 1  26M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.14.fa.gz
-rw-rw-r-- 1  29M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.13.fa.gz
-rw-rw-r-- 1  11M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz
-rw-rw-r-- 1  12M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz
-rw-rw-r-- 1  18M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.20.fa.gz
-rw-rw-r-- 1  67M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz
-rw-rw-r-- 1  16M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.19.fa.gz
-rw-rw-r-- 1  55M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.4.fa.gz
-rw-rw-r-- 1  57M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz
-rw-rw-r-- 1  70M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz
-rw-rw-r-- 1  46M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz
-rw-rw-r-- 1  50M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.6.fa.gz
-rw-rw-r-- 1  52M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz
-rw-rw-r-- 1 6.8M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.Y.fa.gz
-rw-rw-r-- 1  45M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.X.fa.gz
-rw-rw-r-- 1 5.4K Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.MT.fa.gz
-rw-rw-r-- 1  35M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.9.fa.gz
-rw-rw-r-- 1  42M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.8.fa.gz
-rw-rw-r-- 1 3.0G Mar 23 11:21 Homo_sapiens.GRCh38.dna.chromosome.all.fa
-rw-rw-r-- 1  59M Mar 23 11:23 Homo_sapiens.GRCh38.cdna.all.fa.gz
-rw-rw-r-- 1    0 Mar 23 11:30 IDX.stdout
-rw-rw-r-- 1   23 Mar 23 11:30 IDX.stderr
-rw-rw-r-- 1 700M Mar 23 11:31 GRCh38.82.4.bt2
-rw-rw-r-- 1 8.8K Mar 23 11:31 GRCh38.82.3.bt2
-rw-rw-r-- 1 700M Mar 23 11:51 GRCh38.82.2.bt2
-rw-rw-r-- 1 937M Mar 23 11:51 GRCh38.82.1.bt2
-rw-rw-r-- 1    0 Mar 23 11:51 GRCh38.82.rev.2.bt2
-rw-rw-r-- 1 8.0K Mar 23 11:52 GRCh38.82.rev.1.bt2
-rwxr-xr-x 1  956 Mar 23 11:58 indexer.sh
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ cd ~/scratch/Programs/
[rsh46@login-sand3 Programs]$ ls
bamtools                      erne-2.1.1-linux        local                        seqtk
bbmap                         FastQC                  manta-0.29.6.centos5_x86_64  snpEff
bcftools-1.3.1                fastq_screen_v0.5.2     MultiQC                      STAR
Bismark                       gmap-2017-03-17         picard-tools-2.2.4           subread-1.5.0-p2-Linux-x86_64
bowtie2-2.2.9                 hisat2-2.0.4            preseq_v2.0                  svtools
clusterflow                   HTSeq-0.6.1p1           pyBigWig                     telseq
cramtools-3.0                 htslib-1.3.1            pysam                        tophat-2.1.1.Linux_x86_64
cufflinks-2.2.1.Linux_x86_64  infernal-1.1.2          qualimap_v2.2                trim_galore_zip
deepTools                     kallisto_linux-v0.42.5  rrna_mt_count
Drop-seq_tools-1.12           kallisto_linux-v0.43.0  Salmon-0.8.2_linux_x86_64
dupRadar                      kallisto_linux-v0.43.1  samtools-1.3.1
[rsh46@login-sand3 Programs]$ ls gmap-2017-03-17/
acinclude.m4  bin        config.log     configure     INSTALL   Makefile.am  NOTICE  tests  VERSION
aclocal.m4    ChangeLog  config.site    configure.ac  libtool   Makefile.in  README  TODO
AUTHORS       config     config.status  COPYING       Makefile  NEWS         src     util
[rsh46@login-sand3 Programs]$ ls gmap-2017-03-17/bin/
atoiindex      gff3_genes        gmapl            gmap_uncompress  gtf_genes        md_coords        uniqscan
cmetindex      gff3_introns      gmapl.nosimd     gsnap            gtf_introns      psl_genes        uniqscanl
cpuid          gff3_splicesites  gmapl.sse42      gsnapl           gtf_splicesites  psl_introns      vcf_iit
dbsnp_iit      gmap              gmap.nosimd      gsnapl.nosimd    gvf_iit          psl_splicesites
ensembl_genes  gmap_build        gmap_process     gsnapl.sse42     iit_dump         sam_sort
fa_coords      gmap_compress     gmap_reassemble  gsnap.nosimd     iit_get          share
get-genome     gmapindex         gmap.sse42       gsnap.sse42      iit_store        snpindex
[rsh46@login-sand3 Programs]$ vi ~/.bashrc 
[rsh46@login-sand3 Programs]$ cd gmap-2017-03-17/
[rsh46@login-sand3 gmap-2017-03-17]$ ls
acinclude.m4  bin        config.log     configure     INSTALL   Makefile.am  NOTICE  tests  VERSION
aclocal.m4    ChangeLog  config.site    configure.ac  libtool   Makefile.in  README  TODO
AUTHORS       config     config.status  COPYING       Makefile  NEWS         src     util
[rsh46@login-sand3 gmap-2017-03-17]$ cd bin/
[rsh46@login-sand3 bin]$ ls
atoiindex      gff3_genes        gmapl            gmap_uncompress  gtf_genes        md_coords        uniqscan
cmetindex      gff3_introns      gmapl.nosimd     gsnap            gtf_introns      psl_genes        uniqscanl
cpuid          gff3_splicesites  gmapl.sse42      gsnapl           gtf_splicesites  psl_introns      vcf_iit
dbsnp_iit      gmap              gmap.nosimd      gsnapl.nosimd    gvf_iit          psl_splicesites
ensembl_genes  gmap_build        gmap_process     gsnapl.sse42     iit_dump         sam_sort
fa_coords      gmap_compress     gmap_reassemble  gsnap.nosimd     iit_get          share
get-genome     gmapindex         gmap.sse42       gsnap.sse42      iit_store        snpindex
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ du -shc *
4.2M	atoiindex
4.2M	cmetindex
8.0K	cpuid
8.0K	dbsnp_iit
4.0K	ensembl_genes
20K	fa_coords
1.4M	get-genome
8.0K	gff3_genes
16K	gff3_introns
16K	gff3_splicesites
12K	gmap
16K	gmap_build
12K	gmap_compress
3.7M	gmapindex
12K	gmapl
9.3M	gmapl.nosimd
8.7M	gmapl.sse42
9.3M	gmap.nosimd
8.0K	gmap_process
4.0K	gmap_reassemble
8.7M	gmap.sse42
20K	gmap_uncompress
12K	gsnap
12K	gsnapl
9.8M	gsnapl.nosimd
9.2M	gsnapl.sse42
9.8M	gsnap.nosimd
9.2M	gsnap.sse42
8.0K	gtf_genes
12K	gtf_introns
16K	gtf_splicesites
4.0K	gvf_iit
168K	iit_dump
180K	iit_get
120K	iit_store
32K	md_coords
4.0K	psl_genes
8.0K	psl_introns
8.0K	psl_splicesites
208K	sam_sort
12K	share
3.6M	snpindex
9.7M	uniqscan
9.7M	uniqscanl
4.0K	vcf_iit
111M	total
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ mkdir bin
[rsh46@login-sand3 bin]$ ls
atoiindex      get-genome        gmapindex        gmap.sse42       gsnap.sse42      iit_store        snpindex
bin            gff3_genes        gmapl            gmap_uncompress  gtf_genes        md_coords        uniqscan
cmetindex      gff3_introns      gmapl.nosimd     gsnap            gtf_introns      psl_genes        uniqscanl
cpuid          gff3_splicesites  gmapl.sse42      gsnapl           gtf_splicesites  psl_introns      vcf_iit
dbsnp_iit      gmap              gmap.nosimd      gsnapl.nosimd    gvf_iit          psl_splicesites
ensembl_genes  gmap_build        gmap_process     gsnapl.sse42     iit_dump         sam_sort
fa_coords      gmap_compress     gmap_reassemble  gsnap.nosimd     iit_get          share
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ 
[rsh46@login-sand3 bin]$ cd bin
[rsh46@login-sand3 bin]$ ln -s ../gmapindex 
[rsh46@login-sand3 bin]$ cd ~/scratch/Genomes/Homo_sapiens/GRCh38.82/
[rsh46@login-sand3 GRCh38.82]$ ls
GRCh38.82.1.bt2                              Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz
GRCh38.82.2.bt2                              Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz
GRCh38.82.3.bt2                              Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz
GRCh38.82.4.bt2                              Homo_sapiens.GRCh38.dna.chromosome.4.fa.gz
GRCh38.82.rev.1.bt2                          Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz
GRCh38.82.rev.2.bt2                          Homo_sapiens.GRCh38.dna.chromosome.6.fa.gz
Homo_sapiens.GRCh38.82.gtf                   Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz
Homo_sapiens.GRCh38.cdna.all.fa.gz           Homo_sapiens.GRCh38.dna.chromosome.8.fa.gz
Homo_sapiens.GRCh38.cdna.all.kallisto.idx    Homo_sapiens.GRCh38.dna.chromosome.9.fa.gz
Homo_sapiens.GRCh38.cdna.all.salmon.idx      Homo_sapiens.GRCh38.dna.chromosome.all.fa
Homo_sapiens.GRCh38.dna.chromosome.10.fa.gz  Homo_sapiens.GRCh38.dna.chromosome.MT.fa.gz
Homo_sapiens.GRCh38.dna.chromosome.11.fa.gz  Homo_sapiens.GRCh38.dna.chromosome.X.fa.gz
Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz  Homo_sapiens.GRCh38.dna.chromosome.Y.fa.gz
Homo_sapiens.GRCh38.dna.chromosome.13.fa.gz  IDX2.stderr
Homo_sapiens.GRCh38.dna.chromosome.14.fa.gz  IDX2.stdout
Homo_sapiens.GRCh38.dna.chromosome.15.fa.gz  IDX3.stderr
Homo_sapiens.GRCh38.dna.chromosome.16.fa.gz  IDX3.stdout
Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz  IDX.stderr
Homo_sapiens.GRCh38.dna.chromosome.18.fa.gz  IDX.stdout
Homo_sapiens.GRCh38.dna.chromosome.19.fa.gz  indexer.sh
Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz   Log.out
Homo_sapiens.GRCh38.dna.chromosome.20.fa.gz  _STARtmp
Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ ls _STARtmp/
[rsh46@login-sand3 GRCh38.82]$ more Log.out 
STAR version=STAR_2.5.1b_modified
STAR compilation time,server,dir=Mon Apr 18 16:03:54 EDT 2016 florence.cshl.edu:/sonas-hs/gingeras/nlsas_norepl/user/dob
in/STAR/STAR.sandbox/source
##### DEFAULT parameters:
versionSTAR                       20201
versionGenome                     20101   20200   
parametersFiles                   -   
sysShell                          -
runMode                           alignReads
runThreadN                        1
runDirPerm                        User_RWX
runRNGseed                        777
genomeDir                         ./GenomeDir/
genomeLoad                        NoSharedMemory
genomeFastaFiles                  -   
genomeSAindexNbases               14
genomeChrBinNbits                 18
genomeSAsparseD                   1
genomeSuffixLengthMax             18446744073709551615
readFilesIn                       Read1   Read2   
readFilesCommand                  -   
readMatesLengthsIn                NotEqual
readMapNumber                     18446744073709551615
readNameSeparator                 /   
inputBAMfile                      -
bamRemoveDuplicatesType           -
bamRemoveDuplicatesMate2basesN    0
limitGenomeGenerateRAM            31000000000
limitIObufferSize                 150000000
limitOutSAMoneReadBytes           100000
limitOutSJcollapsed               1000000
limitOutSJoneRead                 1000
limitBAMsortRAM                   0
limitSjdbInsertNsj                1000000
outFileNamePrefix                 ./
outTmpDir                         -
outTmpKeep                        None
outStd                            Log
outReadsUnmapped                  None
outQSconversionAdd                0
outMultimapperOrder               Old_2.4
outSAMtype                        SAM   
outSAMmode                        Full
outSAMstrandField                 None
outSAMattributes                  Standard   
outSAMunmapped                    None   
outSAMorder                       Paired
outSAMprimaryFlag                 OneBestScore
outSAMreadID                      Standard
outSAMmapqUnique                  255
outSAMflagOR                      0
outSAMflagAND                     65535
outSAMattrRGline                  -   
outSAMheaderHD                    -   
outSAMheaderPG                    -   
outSAMheaderCommentFile           -
outBAMcompression                 1
outBAMsortingThreadN              0
outSAMfilter                      None   
outSAMmultNmax                    18446744073709551615
outSAMattrIHstart                 1
outSJfilterReads                  All
outSJfilterCountUniqueMin         3   1   1   1   
outSJfilterCountTotalMin          3   1   1   1   
outSJfilterOverhangMin            30   12   12   12   
outSJfilterDistToOtherSJmin       10   0   5   10   
outSJfilterIntronMaxVsReadN       50000   100000   200000   
outWigType                        None   
outWigStrand                      Stranded   
outWigReferencesPrefix            -
outWigNorm                        RPM   
outFilterType                     Normal
outFilterMultimapNmax             10
outFilterMultimapScoreRange       1
outFilterScoreMin                 0
outFilterScoreMinOverLread        0.66
outFilterMatchNmin                0
outFilterMatchNminOverLread       0.66
outFilterMismatchNmax             10
outFilterMismatchNoverLmax        0.3
outFilterMismatchNoverReadLmax    1
outFilterIntronMotifs             None
clip5pNbases                      0   
clip3pNbases                      0   
clip3pAfterAdapterNbases          0   
clip3pAdapterSeq                  -   
clip3pAdapterMMp                  0.1   
winBinNbits                       16
winAnchorDistNbins                9
winFlankNbins                     4
winAnchorMultimapNmax             50
scoreGap                          0
scoreGapNoncan                    -8
scoreGapGCAG                      -4
scoreGapATAC                      -8
scoreStitchSJshift                1
scoreGenomicLengthLog2scale       -0.25
scoreDelBase                      -2
scoreDelOpen                      -2
scoreInsOpen                      -2
scoreInsBase                      -2
seedSearchLmax                    0
seedSearchStartLmax               50
seedSearchStartLmaxOverLread      1
seedPerReadNmax                   1000
seedPerWindowNmax                 50
seedNoneLociPerWindow             10
seedMultimapNmax                  10000
alignIntronMin                    21
alignIntronMax                    0
alignMatesGapMax                  0
alignTranscriptsPerReadNmax       10000
alignSJoverhangMin                5
alignSJDBoverhangMin              3
alignSJstitchMismatchNmax         0   -1   0   0   
alignSplicedMateMapLmin           0
alignSplicedMateMapLminOverLmate    0.66
alignWindowsPerReadNmax           10000
alignTranscriptsPerWindowNmax     100
alignEndsType                     Local
alignSoftClipAtReferenceEnds      Yes
alignEndsProtrude                 0   ConcordantPair   
chimSegmentMin                    0
chimScoreMin                      0
chimScoreDropMax                  20
chimScoreSeparation               10
chimScoreJunctionNonGTAG          -1
chimJunctionOverhangMin           20
chimOutType                       SeparateSAMold
chimFilter                        banGenomicN   
chimSegmentReadGapMax             0
sjdbFileChrStartEnd               -   
sjdbGTFfile                       -
sjdbGTFchrPrefix                  -
sjdbGTFfeatureExon                exon
sjdbGTFtagExonParentTranscript    transcript_id
sjdbGTFtagExonParentGene          gene_id
sjdbOverhang                      100
sjdbScore                         2
sjdbInsertSave                    Basic
quantMode                         -   
quantTranscriptomeBAMcompression    1
quantTranscriptomeBan             IndelSoftclipSingleend
twopass1readsN                    18446744073709551615
twopassMode                       None
##### Command Line:
STAR --runMode genomeGenerate --genomeDir STAR --genomeFastaFiles Homo_sapiens.GRCh38.dna.chromosome.all.fa --runThreadN
 6
##### Initial USER parameters from Command Line:
###### All USER parameters from Command Line:
runMode                       genomeGenerate     ~RE-DEFINED
genomeDir                     STAR     ~RE-DEFINED
genomeFastaFiles              Homo_sapiens.GRCh38.dna.chromosome.all.fa        ~RE-DEFINED
runThreadN                    6     ~RE-DEFINED
##### Finished reading parameters from all sources

##### Final user re-defined parameters-----------------:
runMode                           genomeGenerate
runThreadN                        6
genomeDir                         STAR
genomeFastaFiles                  Homo_sapiens.GRCh38.dna.chromosome.all.fa   

-------------------------------
##### Final effective command line:
STAR   --runMode genomeGenerate   --runThreadN 6   --genomeDir STAR   --genomeFastaFiles Homo_sapiens.GRCh38.dna.chromos
ome.all.fa   

##### Final parameters after user input--------------------------------:
versionSTAR                       20201
versionGenome                     20101   20200   
parametersFiles                   -   
sysShell                          -
runMode                           genomeGenerate
runThreadN                        6
runDirPerm                        User_RWX
runRNGseed                        777
genomeDir                         STAR
genomeLoad                        NoSharedMemory
genomeFastaFiles                  Homo_sapiens.GRCh38.dna.chromosome.all.fa   
genomeSAindexNbases               14
genomeChrBinNbits                 18
genomeSAsparseD                   1
genomeSuffixLengthMax             18446744073709551615
readFilesIn                       Read1   Read2   
readFilesCommand                  -   
readMatesLengthsIn                NotEqual
readMapNumber                     18446744073709551615
readNameSeparator                 /   
inputBAMfile                      -
bamRemoveDuplicatesType           -
bamRemoveDuplicatesMate2basesN    0
limitGenomeGenerateRAM            31000000000
limitIObufferSize                 150000000
limitOutSAMoneReadBytes           100000
limitOutSJcollapsed               1000000
limitOutSJoneRead                 1000
limitBAMsortRAM                   0
limitSjdbInsertNsj                1000000
outFileNamePrefix                 ./
outTmpDir                         -
outTmpKeep                        None
outStd                            Log
outReadsUnmapped                  None
outQSconversionAdd                0
outMultimapperOrder               Old_2.4
outSAMtype                        SAM   
outSAMmode                        Full
outSAMstrandField                 None
outSAMattributes                  Standard   
outSAMunmapped                    None   
outSAMorder                       Paired
outSAMprimaryFlag                 OneBestScore
outSAMreadID                      Standard
outSAMmapqUnique                  255
outSAMflagOR                      0
outSAMflagAND                     65535
outSAMattrRGline                  -   
outSAMheaderHD                    -   
outSAMheaderPG                    -   
outSAMheaderCommentFile           -
outBAMcompression                 1
outBAMsortingThreadN              0
outSAMfilter                      None   
outSAMmultNmax                    18446744073709551615
outSAMattrIHstart                 1
outSJfilterReads                  All
outSJfilterCountUniqueMin         3   1   1   1   
outSJfilterCountTotalMin          3   1   1   1   
outSJfilterOverhangMin            30   12   12   12   
outSJfilterDistToOtherSJmin       10   0   5   10   
outSJfilterIntronMaxVsReadN       50000   100000   200000   
outWigType                        None   
outWigStrand                      Stranded   
outWigReferencesPrefix            -
outWigNorm                        RPM   
outFilterType                     Normal
outFilterMultimapNmax             10
outFilterMultimapScoreRange       1
outFilterScoreMin                 0
outFilterScoreMinOverLread        0.66
outFilterMatchNmin                0
outFilterMatchNminOverLread       0.66
outFilterMismatchNmax             10
outFilterMismatchNoverLmax        0.3
outFilterMismatchNoverReadLmax    1
outFilterIntronMotifs             None
clip5pNbases                      0   
clip3pNbases                      0   
clip3pAfterAdapterNbases          0   
clip3pAdapterSeq                  -   
clip3pAdapterMMp                  0.1   
winBinNbits                       16
winAnchorDistNbins                9
winFlankNbins                     4
winAnchorMultimapNmax             50
scoreGap                          0
scoreGapNoncan                    -8
scoreGapGCAG                      -4
scoreGapATAC                      -8
scoreStitchSJshift                1
scoreGenomicLengthLog2scale       -0.25
scoreDelBase                      -2
scoreDelOpen                      -2
scoreInsOpen                      -2
scoreInsBase                      -2
seedSearchLmax                    0
seedSearchStartLmax               50
seedSearchStartLmaxOverLread      1
seedPerReadNmax                   1000
seedPerWindowNmax                 50
seedNoneLociPerWindow             10
seedMultimapNmax                  10000
alignIntronMin                    21
alignIntronMax                    0
alignMatesGapMax                  0
alignTranscriptsPerReadNmax       10000
alignSJoverhangMin                5
alignSJDBoverhangMin              3
alignSJstitchMismatchNmax         0   -1   0   0   
alignSplicedMateMapLmin           0
alignSplicedMateMapLminOverLmate    0.66
alignWindowsPerReadNmax           10000
alignTranscriptsPerWindowNmax     100
alignEndsType                     Local
alignSoftClipAtReferenceEnds      Yes
alignEndsProtrude                 0   ConcordantPair   
chimSegmentMin                    0
chimScoreMin                      0
chimScoreDropMax                  20
chimScoreSeparation               10
chimScoreJunctionNonGTAG          -1
chimJunctionOverhangMin           20
chimOutType                       SeparateSAMold
chimFilter                        banGenomicN   
chimSegmentReadGapMax             0
sjdbFileChrStartEnd               -   
sjdbGTFfile                       -
sjdbGTFchrPrefix                  -
sjdbGTFfeatureExon                exon
sjdbGTFtagExonParentTranscript    transcript_id
sjdbGTFtagExonParentGene          gene_id
sjdbOverhang                      100
sjdbScore                         2
sjdbInsertSave                    Basic
quantMode                         -   
quantTranscriptomeBAMcompression    1
quantTranscriptomeBan             IndelSoftclipSingleend
twopass1readsN                    18446744073709551615
twopassMode                       None
----------------------------------------

Finished loading and checking parameters
Mar 23 12:18:22 ... starting to generate Genome files

genomeGenerate.cpp:150:genomeGenerate: exiting because of *OUTPUT FILE* error: could not create output file STAR/genomeP
arameters.txt
Solution: check that the path exists and you have write permission for this file

Mar 23 12:18:22 ...... FATAL ERROR, exiting
[rsh46@login-sand3 GRCh38.82]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601770   IDX2       rsh46         Running 6       23:10:51  Thu Mar 23 12:00:40
5601891   IDX        rsh46         Running 6       23:59:15  Thu Mar 23 12:49:04

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================

Total Jobs: 2     Active Jobs: 2     Idle Jobs: 0     Blocked Jobs: 0   
[rsh46@login-sand3 GRCh38.82]$ ls -hltrog
total 12G
-rw-rw-r-- 1 1.4G Mar 23 11:16 Homo_sapiens.GRCh38.82.gtf
-rw-rw-r-- 1  38M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.11.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.10.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.18.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz
-rw-rw-r-- 1  24M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.16.fa.gz
-rw-rw-r-- 1  25M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.15.fa.gz
-rw-rw-r-- 1  26M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.14.fa.gz
-rw-rw-r-- 1  29M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.13.fa.gz
-rw-rw-r-- 1  11M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz
-rw-rw-r-- 1  12M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz
-rw-rw-r-- 1  18M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.20.fa.gz
-rw-rw-r-- 1  67M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz
-rw-rw-r-- 1  16M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.19.fa.gz
-rw-rw-r-- 1  55M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.4.fa.gz
-rw-rw-r-- 1  57M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz
-rw-rw-r-- 1  70M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz
-rw-rw-r-- 1  46M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz
-rw-rw-r-- 1  50M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.6.fa.gz
-rw-rw-r-- 1  52M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz
-rw-rw-r-- 1 6.8M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.Y.fa.gz
-rw-rw-r-- 1  45M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.X.fa.gz
-rw-rw-r-- 1 5.4K Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.MT.fa.gz
-rw-rw-r-- 1  35M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.9.fa.gz
-rw-rw-r-- 1  42M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.8.fa.gz
-rw-rw-r-- 1 3.0G Mar 23 11:21 Homo_sapiens.GRCh38.dna.chromosome.all.fa
-rw-rw-r-- 1  59M Mar 23 11:23 Homo_sapiens.GRCh38.cdna.all.fa.gz
-rw-rw-r-- 1 700M Mar 23 11:31 GRCh38.82.4.bt2
-rw-rw-r-- 1 8.8K Mar 23 11:31 GRCh38.82.3.bt2
-rw-rw-r-- 1 700M Mar 23 11:51 GRCh38.82.2.bt2
-rw-rw-r-- 1 937M Mar 23 11:51 GRCh38.82.1.bt2
-rw-rw-r-- 1  696 Mar 23 12:00 IDX2.stderr
-rw-rw-r-- 1   11 Mar 23 12:10 IDX3.stdout
-rw-rw-r-- 1  975 Mar 23 12:10 IDX3.stderr
-rw-rw-r-- 1 700M Mar 23 12:11 GRCh38.82.rev.2.bt2
-rw-rw-r-- 1 937M Mar 23 12:11 GRCh38.82.rev.1.bt2
drwxrwxr-x 2 4.0K Mar 23 12:14 Homo_sapiens.GRCh38.cdna.all.salmon.idx
-rw-rw-r-- 1 2.2G Mar 23 12:18 Homo_sapiens.GRCh38.cdna.all.kallisto.idx
-rw-rw-r-- 1 4.7K Mar 23 12:48 IDX2.stdout
-rwxr-xr-x 1 1.1K Mar 23 12:49 indexer.sh
drwx------ 2 4.0K Mar 23 12:49 _STARtmp
drwxrwxr-x 2 4.0K Mar 23 12:49 STAR
-rw-rw-r-- 1   93 Mar 23 12:49 IDX.stdout
-rw-rw-r-- 1    0 Mar 23 12:49 IDX.stderr
-rw-rw-r-- 1  15K Mar 23 12:49 Log.out
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ ls -hltrog
total 12G
-rw-rw-r-- 1 1.4G Mar 23 11:16 Homo_sapiens.GRCh38.82.gtf
-rw-rw-r-- 1  38M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.11.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.10.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.18.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz
-rw-rw-r-- 1  24M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.16.fa.gz
-rw-rw-r-- 1  25M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.15.fa.gz
-rw-rw-r-- 1  26M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.14.fa.gz
-rw-rw-r-- 1  29M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.13.fa.gz
-rw-rw-r-- 1  11M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz
-rw-rw-r-- 1  12M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz
-rw-rw-r-- 1  18M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.20.fa.gz
-rw-rw-r-- 1  67M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz
-rw-rw-r-- 1  16M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.19.fa.gz
-rw-rw-r-- 1  55M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.4.fa.gz
-rw-rw-r-- 1  57M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz
-rw-rw-r-- 1  70M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz
-rw-rw-r-- 1  46M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz
-rw-rw-r-- 1  50M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.6.fa.gz
-rw-rw-r-- 1  52M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz
-rw-rw-r-- 1 6.8M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.Y.fa.gz
-rw-rw-r-- 1  45M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.X.fa.gz
-rw-rw-r-- 1 5.4K Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.MT.fa.gz
-rw-rw-r-- 1  35M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.9.fa.gz
-rw-rw-r-- 1  42M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.8.fa.gz
-rw-rw-r-- 1 3.0G Mar 23 11:21 Homo_sapiens.GRCh38.dna.chromosome.all.fa
-rw-rw-r-- 1  59M Mar 23 11:23 Homo_sapiens.GRCh38.cdna.all.fa.gz
-rw-rw-r-- 1 700M Mar 23 11:31 GRCh38.82.4.bt2
-rw-rw-r-- 1 8.8K Mar 23 11:31 GRCh38.82.3.bt2
-rw-rw-r-- 1 700M Mar 23 11:51 GRCh38.82.2.bt2
-rw-rw-r-- 1 937M Mar 23 11:51 GRCh38.82.1.bt2
-rw-rw-r-- 1  696 Mar 23 12:00 IDX2.stderr
-rw-rw-r-- 1   11 Mar 23 12:10 IDX3.stdout
-rw-rw-r-- 1  975 Mar 23 12:10 IDX3.stderr
-rw-rw-r-- 1 700M Mar 23 12:11 GRCh38.82.rev.2.bt2
-rw-rw-r-- 1 937M Mar 23 12:11 GRCh38.82.rev.1.bt2
drwxrwxr-x 2 4.0K Mar 23 12:14 Homo_sapiens.GRCh38.cdna.all.salmon.idx
-rw-rw-r-- 1 2.2G Mar 23 12:18 Homo_sapiens.GRCh38.cdna.all.kallisto.idx
-rw-rw-r-- 1 4.7K Mar 23 12:48 IDX2.stdout
drwx------ 2 4.0K Mar 23 12:49 _STARtmp
-rw-rw-r-- 1    0 Mar 23 12:49 IDX.stderr
drwxrwxr-x 2 4.0K Mar 23 12:49 STAR
-rw-rw-r-- 1  15K Mar 23 12:50 Log.out
-rw-rw-r-- 1  248 Mar 23 12:50 IDX.stdout
-rwxr-xr-x 1 1.2K Mar 23 12:51 indexer.sh
[rsh46@login-sand3 GRCh38.82]$ cd ~/scratch/Programs/
[rsh46@login-sand3 Programs]$ ls
bamtools                      erne-2.1.1-linux        local                        seqtk
bbmap                         FastQC                  manta-0.29.6.centos5_x86_64  snpEff
bcftools-1.3.1                fastq_screen_v0.5.2     MultiQC                      STAR
Bismark                       gmap-2017-03-17         picard-tools-2.2.4           subread-1.5.0-p2-Linux-x86_64
bowtie2-2.2.9                 hisat2-2.0.4            preseq_v2.0                  svtools
clusterflow                   HTSeq-0.6.1p1           pyBigWig                     telseq
cramtools-3.0                 htslib-1.3.1            pysam                        tophat-2.1.1.Linux_x86_64
cufflinks-2.2.1.Linux_x86_64  infernal-1.1.2          qualimap_v2.2                trim_galore_zip
deepTools                     kallisto_linux-v0.42.5  rrna_mt_count
Drop-seq_tools-1.12           kallisto_linux-v0.43.0  Salmon-0.8.2_linux_x86_64
dupRadar                      kallisto_linux-v0.43.1  samtools-1.3.1
[rsh46@login-sand3 Programs]$ 
[rsh46@login-sand3 Programs]$ 
[rsh46@login-sand3 Programs]$ 
[rsh46@login-sand3 Programs]$ cd gmap-2017-03-17/
[rsh46@login-sand3 gmap-2017-03-17]$ ls
acinclude.m4  bin        config.log     configure     INSTALL   Makefile.am  NOTICE  tests  VERSION
aclocal.m4    ChangeLog  config.site    configure.ac  libtool   Makefile.in  README  TODO
AUTHORS       config     config.status  COPYING       Makefile  NEWS         src     util
[rsh46@login-sand3 gmap-2017-03-17]$ rm -r bin/
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ history | grep config
  301  2017-02-28 09:34:26 multiqc -f -c CTR_acf1004_0002.config.yaml .
  303  2017-02-28 09:34:36 multiqc -f -c CTR_acf1004_0002.config.yaml .
  410  2017-03-01 14:16:12 vi ~/scratch/Programs/clusterflow/pipelines/fastq_tophat.config 
  419  2017-03-01 14:17:38 vi fastq_tophat_double.config
  440  2017-03-02 09:52:36 more pipelines/fastq_tophat_double.config 
  499  2017-03-08 16:51:15 more /etc/sysconfig/slurm
  692  2017-03-22 12:41:34 ./configure --prefix=/home/rsh46/scratch/Programs/gmap-2017-03-17/bin
  722  2017-03-22 13:04:28 more /scratch/rsh46/Programs/clusterflow/pipelines/fastq_hisat2.config 
  732  2017-03-22 13:55:41 vi ~/.clusterflow/genomes.config 
  750  2017-03-22 15:39:20 vi ~/.clusterflow/genomes.config 
 1066  2017-03-23 12:53:15 history | grep config
[rsh46@login-sand3 gmap-2017-03-17]$ ./configure --prefix=/home/rsh46/scratch/Programs/gmap-2017-03-17/
checking package version... 2017-03-17
loading default site script ./config.site
checking CFLAGS... not set by user so using default -O3 -fomit-frame-pointer
checking MPI_CFLAGS... 
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for non-Intel CPU... no
checking for gcc... /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc accepts -g... yes
checking for /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc option to accept ISO C89... none needed
checking whether /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc understands -c and -o together... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking whether make supports nested variables... yes
checking dependency style of /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc... gcc3
checking whether to enable maintainer-specific portions of Makefiles... no
checking bindir... /home/rsh46/scratch/Programs/gmap-2017-03-17/bin
checking for a working version of perl... /usr/bin/perl
checking for gcc... (cached) /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc accepts -g... (cached) yes
checking for /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc option to accept ISO C89... (cached) none needed
checking whether /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc understands -c and -o together... (cached) yes
checking for C compiler vendor... gnu
checking for mpicc... /usr/local/Cluster-Apps/intel/impi/4.1.3.045/bin64/mpicc
checking for MPI_Init... yes
checking for mpi.h... yes
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for ld used by /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1966080
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... mt
checking if mt is a manifest tool... no
checking how to run the C preprocessor... /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc supports -fno-rtti -fno-exceptions... no
checking for /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc option to produce PIC... -fPIC -DPIC
checking if /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc PIC flag -fPIC -DPIC works... yes
checking if /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc static flag -static works... yes
checking if /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc supports -c -o file.o... yes
checking if /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc supports -c -o file.o... (cached) yes
checking whether the /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking for rint in -lm... yes
checking whether to use MPI_File for input... not specified so enabled by default
checking for pthreads feature... not specified so enabled by default
checking for the pthreads library -lpthreads... no
checking whether pthreads work without any flags... no
checking whether pthreads work with -Kthread... no
checking whether pthreads work with -kthread... no
checking for the pthreads library -llthread... no
checking whether pthreads work with -pthread... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking if more special flags are required for pthreads... no
checking for cc_r... /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc
checking for ANSI C header files... (cached) yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking stddef.h usability... yes
checking stddef.h presence... yes
checking for stddef.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for strings.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/types.h... (cached) yes
checking whether byte ordering is bigendian... no
checking for an ANSI C-conforming const... yes
checking for working volatile... yes
checking for size_t... yes
checking for off_t... yes
checking for caddr_t... yes
checking size of unsigned long... 8
checking size of unsigned long long... 8
checking size of off_t... 8
checking for _LARGEFILE_SOURCE value needed for large files... no
checking whether alloca is enabled... not specified so enabled by default
checking for working alloca.h... yes
checking for alloca... yes
checking whether mmap is enabled... not specified so enabled by default
checking for working mmap with MAP_FIXED... yes
checking for working mmap with MAP_VARIABLE... no
checking for MAP_FILE in mmap... yes
checking for MAP_VARIABLE in mmap... no
checking for MAP_SHARED in mmap... yes
checking for MAP_PRIVATE in mmap... yes
checking for MAP_FAILED in mmap... yes
checking for MADV_DONTNEED in madvise... yes
checking for MADV_WILLNEED in madvise... yes
checking for MADV_RANDOM in madvise... yes
checking for MADV_SEQUENTIAL in madvise... yes
checking for SHM_NORESERVE in shmget... yes
checking for ceil... yes
checking for floor... yes
checking for index... yes
checking for log... yes
checking for madvise... yes
checking for memcpy... yes
checking for memmove... yes
checking for memset... yes
checking for munmap... yes
checking for pow... yes
checking for rint... yes
checking for stat64... yes
checking for strtoul... yes
checking for sysconf... yes
checking for sysctl... yes
checking for sigaction... yes
checking for shmget... yes
checking for shmctl... yes
checking for shmat... yes
checking for shmdt... yes
checking for semget... yes
checking for semctl... yes
checking for semop... yes
checking for struct stat64... no
checking for pagesize via sysconf... yes
checking for pagesize via sysctl... no
checking whether fopen accepts "b" mode... yes
checking whether fopen accepts "t" mode... yes
checking whether C compiler accepts -O3 -fomit-frame-pointer -mpopcnt... yes
checking for __builtin_popcount... yes
checking for __builtin_clz... yes
checking for __builtin_ctz... yes
checking for bsr instruction in assembly... yes
checking compiler is defective and requires an immediate in sse2 shift commands... no
checking for whether simd is enabled... not specified so enabled by default
checking for sse2 support... yes
checking for ssse3 support... yes
checking for sse4.1 support... yes
checking for sse4.2 support... yes
checking for popcnt support... yes
checking for bmi1 support... no
checking for avx2 support... no
checking for bmi2 support... no
checking for avx512 support... no
checking whether C compiler accepts -msse2... yes
checking for emmintrin.h header file... yes
checking whether C compiler accepts -msse2 -mssse3... yes
checking for tmmintrin.h header file... yes
checking whether C compiler accepts -msse2 -mssse3 -msse4.1... yes
checking for smmintrin.h header file... yes
checking whether C compiler accepts -msse2 -mssse3 -msse4.1 -msse4.2... yes
checking for nmmintrin.h header file... yes
checking for mm_popcnt support... yes
checking gmapdb... /home/rsh46/scratch/Programs/gmap-2017-03-17/share
checking MAX_STACK_READLENGTH... 300
checking for zlib support... enabled
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for gzopen in -lz... yes
checking for gzeof in -lz... yes
checking for gzgetc in -lz... yes
checking for gzgets in -lz... yes
checking for gzclose in -lz... yes
checking if zlib package is complete... working
checking for gzbuffer in -lz... no
checking for bzlib support... enabled
checking bzlib.h usability... yes
checking bzlib.h presence... yes
checking for bzlib.h... yes
checking for BZ2_bzReadOpen in -lbz2... yes
checking for BZ2_bzRead in -lbz2... yes
checking for BZ2_bzReadClose in -lbz2... yes
checking if bzlib package is complete... working
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating util/Makefile
config.status: creating util/gmap_compress.pl
config.status: creating util/gmap_uncompress.pl
config.status: creating util/gmap_process.pl
config.status: creating util/gmap_build.pl
config.status: creating util/gmap_reassemble.pl
config.status: creating util/md_coords.pl
config.status: creating util/fa_coords.pl
config.status: creating util/psl_splicesites.pl
config.status: creating util/psl_introns.pl
config.status: creating util/psl_genes.pl
config.status: creating util/ensembl_genes.pl
config.status: creating util/gtf_splicesites.pl
config.status: creating util/gtf_introns.pl
config.status: creating util/gtf_genes.pl
config.status: creating util/gff3_splicesites.pl
config.status: creating util/gff3_introns.pl
config.status: creating util/gff3_genes.pl
config.status: creating util/dbsnp_iit.pl
config.status: creating util/gvf_iit.pl
config.status: creating util/vcf_iit.pl
config.status: creating tests/Makefile
config.status: creating tests/align.test
config.status: creating tests/coords1.test
config.status: creating tests/setup1.test
config.status: creating tests/iit.test
config.status: creating src/config.h
config.status: src/config.h is unchanged
config.status: executing depfiles commands
config.status: executing libtool commands
checking standard compiler and flags to be used... /usr/local/Cluster-Apps/gcc/5.2.0/bin/gcc -O3 -fomit-frame-pointer
checking standard linker flags to be used... 
checking pthread compiler flags to be used... -pthread
checking popcnt flags to be used... -mpopcnt
checking compile level... sse42
checking SSE4.2 compiler flags to be used... -msse2 -mssse3 -msse4.1 -msse4.2
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ make
Making all in src
make[1]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17/src'
make  all-am
make[2]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17/src'
make[2]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17/src'
make[1]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17/src'
Making all in util
make[1]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17/util'
cp gmap_compress.pl gmap_compress
chmod +x gmap_compress
cp gmap_uncompress.pl gmap_uncompress
chmod +x gmap_uncompress
cp gmap_process.pl gmap_process
chmod +x gmap_process
cp gmap_build.pl gmap_build
chmod +x gmap_build
cp gmap_reassemble.pl gmap_reassemble
chmod +x gmap_reassemble
cp md_coords.pl md_coords
chmod +x md_coords
cp fa_coords.pl fa_coords
chmod +x fa_coords
cp psl_splicesites.pl psl_splicesites
chmod +x psl_splicesites
cp psl_introns.pl psl_introns
chmod +x psl_introns
cp psl_genes.pl psl_genes
chmod +x psl_genes
cp ensembl_genes.pl ensembl_genes
chmod +x ensembl_genes
cp gtf_splicesites.pl gtf_splicesites
chmod +x gtf_splicesites
cp gtf_introns.pl gtf_introns
chmod +x gtf_introns
cp gtf_genes.pl gtf_genes
chmod +x gtf_genes
cp gff3_splicesites.pl gff3_splicesites
chmod +x gff3_splicesites
cp gff3_introns.pl gff3_introns
chmod +x gff3_introns
cp gff3_genes.pl gff3_genes
chmod +x gff3_genes
cp dbsnp_iit.pl dbsnp_iit
chmod +x dbsnp_iit
cp gvf_iit.pl gvf_iit
chmod +x gvf_iit
cp vcf_iit.pl vcf_iit
chmod +x vcf_iit
make[1]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17/util'
Making all in tests
make[1]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17/tests'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17/tests'
make[1]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17'
make[1]: Nothing to be done for `all-am'.
make[1]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17'
[rsh46@login-sand3 gmap-2017-03-17]$ make install
Making install in src
make[1]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17/src'
make[2]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17/src'
 /bin/mkdir -p '/home/rsh46/scratch/Programs/gmap-2017-03-17/bin'
  /bin/sh ../libtool   --mode=install /usr/bin/install -c cpuid gmap gmapl get-genome gmapindex iit_store iit_get iit_dump gsnap gsnapl uniqscan uniqscanl snpindex cmetindex atoiindex sam_sort gmap.nosimd gmapl.nosimd gsnap.nosimd gsnapl.nosimd gmap.sse42 gmapl.sse42 gsnap.sse42 gsnapl.sse42 '/home/rsh46/scratch/Programs/gmap-2017-03-17/bin'
libtool: install: /usr/bin/install -c cpuid /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/cpuid
libtool: install: /usr/bin/install -c gmap /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gmap
libtool: install: /usr/bin/install -c gmapl /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gmapl
libtool: install: /usr/bin/install -c get-genome /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/get-genome
libtool: install: /usr/bin/install -c gmapindex /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gmapindex
libtool: install: /usr/bin/install -c iit_store /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/iit_store
libtool: install: /usr/bin/install -c iit_get /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/iit_get
libtool: install: /usr/bin/install -c iit_dump /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/iit_dump
libtool: install: /usr/bin/install -c gsnap /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gsnap
libtool: install: /usr/bin/install -c gsnapl /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gsnapl
libtool: install: /usr/bin/install -c uniqscan /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/uniqscan
libtool: install: /usr/bin/install -c uniqscanl /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/uniqscanl
libtool: install: /usr/bin/install -c snpindex /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/snpindex
libtool: install: /usr/bin/install -c cmetindex /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/cmetindex
libtool: install: /usr/bin/install -c atoiindex /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/atoiindex
libtool: install: /usr/bin/install -c sam_sort /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/sam_sort
libtool: install: /usr/bin/install -c gmap.nosimd /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gmap.nosimd
libtool: install: /usr/bin/install -c gmapl.nosimd /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gmapl.nosimd
libtool: install: /usr/bin/install -c gsnap.nosimd /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gsnap.nosimd
libtool: install: /usr/bin/install -c gsnapl.nosimd /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gsnapl.nosimd
libtool: install: /usr/bin/install -c gmap.sse42 /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gmap.sse42
libtool: install: /usr/bin/install -c gmapl.sse42 /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gmapl.sse42
libtool: install: /usr/bin/install -c gsnap.sse42 /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gsnap.sse42
libtool: install: /usr/bin/install -c gsnapl.sse42 /home/rsh46/scratch/Programs/gmap-2017-03-17/bin/gsnapl.sse42
make[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17/src'
make[1]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17/src'
Making install in util
make[1]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17/util'
make[2]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17/util'
 /bin/mkdir -p '/home/rsh46/scratch/Programs/gmap-2017-03-17/bin'
 /usr/bin/install -c gmap_compress gmap_uncompress gmap_process gmap_build gmap_reassemble md_coords fa_coords psl_splicesites psl_introns psl_genes ensembl_genes gtf_splicesites gtf_introns gtf_genes gff3_splicesites gff3_introns gff3_genes dbsnp_iit gvf_iit vcf_iit '/home/rsh46/scratch/Programs/gmap-2017-03-17/bin'
make[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17/util'
make[1]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17/util'
Making install in tests
make[1]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17/tests'
make[2]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17/tests'
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17/tests'
make[1]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17/tests'
make[1]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17'
make[2]: Entering directory `/scratch/rsh46/Programs/gmap-2017-03-17'
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
make[2]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17'
make[1]: Leaving directory `/scratch/rsh46/Programs/gmap-2017-03-17'
[rsh46@login-sand3 gmap-2017-03-17]$ ls
acinclude.m4  bin        config.log     configure     INSTALL   Makefile.am  NOTICE  tests  VERSION
aclocal.m4    ChangeLog  config.site    configure.ac  libtool   Makefile.in  README  TODO
AUTHORS       config     config.status  COPYING       Makefile  NEWS         src     util
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ ls bin/
atoiindex      gff3_genes        gmapl            gmap_uncompress  gtf_genes        md_coords        uniqscanl
cmetindex      gff3_introns      gmapl.nosimd     gsnap            gtf_introns      psl_genes        vcf_iit
cpuid          gff3_splicesites  gmapl.sse42      gsnapl           gtf_splicesites  psl_introns
dbsnp_iit      gmap              gmap.nosimd      gsnapl.nosimd    gvf_iit          psl_splicesites
ensembl_genes  gmap_build        gmap_process     gsnapl.sse42     iit_dump         sam_sort
fa_coords      gmap_compress     gmap_reassemble  gsnap.nosimd     iit_get          snpindex
get-genome     gmapindex         gmap.sse42       gsnap.sse42      iit_store        uniqscan
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ pwd
/home/rsh46/scratch/Programs/gmap-2017-03-17
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ 
[rsh46@login-sand3 gmap-2017-03-17]$ cd ~/scratch/Genomes/Homo_sapiens/GRCh38.82/
[rsh46@login-sand3 GRCh38.82]$ ,s
-bash: ,s: command not found
[rsh46@login-sand3 GRCh38.82]$ ls
GRCh38.82.1.bt2                              Homo_sapiens.GRCh38.dna.chromosome.20.fa.gz
GRCh38.82.1.ht2                              Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz
GRCh38.82.2.bt2                              Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz
GRCh38.82.2.ht2                              Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz
GRCh38.82.3.bt2                              Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz
GRCh38.82.3.ht2                              Homo_sapiens.GRCh38.dna.chromosome.4.fa.gz
GRCh38.82.4.bt2                              Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz
GRCh38.82.4.ht2                              Homo_sapiens.GRCh38.dna.chromosome.6.fa.gz
GRCh38.82.7.ht2                              Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz
GRCh38.82.8.ht2                              Homo_sapiens.GRCh38.dna.chromosome.8.fa.gz
GRCh38.82.rev.1.bt2                          Homo_sapiens.GRCh38.dna.chromosome.9.fa.gz
GRCh38.82.rev.2.bt2                          Homo_sapiens.GRCh38.dna.chromosome.all.fa
Homo_sapiens.GRCh38.82.gtf                   Homo_sapiens.GRCh38.dna.chromosome.MT.fa.gz
Homo_sapiens.GRCh38.cdna.all.fa.gz           Homo_sapiens.GRCh38.dna.chromosome.X.fa.gz
Homo_sapiens.GRCh38.cdna.all.kallisto.idx    Homo_sapiens.GRCh38.dna.chromosome.Y.fa.gz
Homo_sapiens.GRCh38.cdna.all.salmon.idx      IDX2.stderr
Homo_sapiens.GRCh38.dna.chromosome.10.fa.gz  IDX2.stdout
Homo_sapiens.GRCh38.dna.chromosome.11.fa.gz  IDX3.stderr
Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz  IDX3.stdout
Homo_sapiens.GRCh38.dna.chromosome.13.fa.gz  IDXHS2.stderr
Homo_sapiens.GRCh38.dna.chromosome.14.fa.gz  IDXHS2.stdout
Homo_sapiens.GRCh38.dna.chromosome.15.fa.gz  IDX.stderr
Homo_sapiens.GRCh38.dna.chromosome.16.fa.gz  IDX.stdout
Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz  indexer.sh
Homo_sapiens.GRCh38.dna.chromosome.18.fa.gz  Log.out
Homo_sapiens.GRCh38.dna.chromosome.19.fa.gz  STAR
Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz   _STARtmp
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ ls -hltrog
total 12G
-rw-rw-r-- 1 1.4G Mar 23 11:16 Homo_sapiens.GRCh38.82.gtf
-rw-rw-r-- 1  38M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.11.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.10.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.18.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz
-rw-rw-r-- 1  24M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.16.fa.gz
-rw-rw-r-- 1  25M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.15.fa.gz
-rw-rw-r-- 1  26M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.14.fa.gz
-rw-rw-r-- 1  29M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.13.fa.gz
-rw-rw-r-- 1  11M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz
-rw-rw-r-- 1  12M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz
-rw-rw-r-- 1  18M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.20.fa.gz
-rw-rw-r-- 1  67M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz
-rw-rw-r-- 1  16M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.19.fa.gz
-rw-rw-r-- 1  55M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.4.fa.gz
-rw-rw-r-- 1  57M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz
-rw-rw-r-- 1  70M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz
-rw-rw-r-- 1  46M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz
-rw-rw-r-- 1  50M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.6.fa.gz
-rw-rw-r-- 1  52M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz
-rw-rw-r-- 1 6.8M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.Y.fa.gz
-rw-rw-r-- 1  45M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.X.fa.gz
-rw-rw-r-- 1 5.4K Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.MT.fa.gz
-rw-rw-r-- 1  35M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.9.fa.gz
-rw-rw-r-- 1  42M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.8.fa.gz
-rw-rw-r-- 1 3.0G Mar 23 11:21 Homo_sapiens.GRCh38.dna.chromosome.all.fa
-rw-rw-r-- 1  59M Mar 23 11:23 Homo_sapiens.GRCh38.cdna.all.fa.gz
-rw-rw-r-- 1 700M Mar 23 11:31 GRCh38.82.4.bt2
-rw-rw-r-- 1 8.8K Mar 23 11:31 GRCh38.82.3.bt2
-rw-rw-r-- 1 700M Mar 23 11:51 GRCh38.82.2.bt2
-rw-rw-r-- 1 937M Mar 23 11:51 GRCh38.82.1.bt2
-rw-rw-r-- 1  696 Mar 23 12:00 IDX2.stderr
-rw-rw-r-- 1   11 Mar 23 12:10 IDX3.stdout
-rw-rw-r-- 1  975 Mar 23 12:10 IDX3.stderr
-rw-rw-r-- 1 700M Mar 23 12:11 GRCh38.82.rev.2.bt2
-rw-rw-r-- 1 937M Mar 23 12:11 GRCh38.82.rev.1.bt2
drwxrwxr-x 2 4.0K Mar 23 12:14 Homo_sapiens.GRCh38.cdna.all.salmon.idx
-rw-rw-r-- 1 2.2G Mar 23 12:18 Homo_sapiens.GRCh38.cdna.all.kallisto.idx
-rw-rw-r-- 1 4.7K Mar 23 12:48 IDX2.stdout
drwx------ 2 4.0K Mar 23 12:49 _STARtmp
drwxrwxr-x 2 4.0K Mar 23 12:49 STAR
-rw-rw-r-- 1  15K Mar 23 12:50 Log.out
-rw-rw-r-- 1  248 Mar 23 12:50 IDX.stdout
-rw-rw-r-- 1    0 Mar 23 12:52 IDXHS2.stderr
-rw-rw-r-- 1    7 Mar 23 12:52 IDXHS2.stdout
-rw-rw-r-- 1 700M Mar 23 12:53 GRCh38.82.4.ht2
-rw-rw-r-- 1 8.8K Mar 23 12:53 GRCh38.82.3.ht2
-rw-rw-r-- 1    8 Mar 23 12:53 GRCh38.82.8.ht2
-rw-rw-r-- 1    8 Mar 23 12:53 GRCh38.82.7.ht2
-rw-rw-r-- 1    4 Mar 23 12:53 GRCh38.82.2.ht2
-rw-rw-r-- 1  264 Mar 23 12:55 IDX.stderr
-rw-rw-r-- 1  12K Mar 23 12:55 GRCh38.82.1.ht2
-rwxr-xr-x 1 1.2K Mar 23 12:55 indexer.sh
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ ls -hltrog
total 13G
-rw-rw-r-- 1 1.4G Mar 23 11:16 Homo_sapiens.GRCh38.82.gtf
-rw-rw-r-- 1  38M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.11.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.10.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.18.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz
-rw-rw-r-- 1  24M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.16.fa.gz
-rw-rw-r-- 1  25M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.15.fa.gz
-rw-rw-r-- 1  26M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.14.fa.gz
-rw-rw-r-- 1  29M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.13.fa.gz
-rw-rw-r-- 1  11M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz
-rw-rw-r-- 1  12M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz
-rw-rw-r-- 1  18M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.20.fa.gz
-rw-rw-r-- 1  67M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz
-rw-rw-r-- 1  16M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.19.fa.gz
-rw-rw-r-- 1  55M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.4.fa.gz
-rw-rw-r-- 1  57M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz
-rw-rw-r-- 1  70M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz
-rw-rw-r-- 1  46M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz
-rw-rw-r-- 1  50M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.6.fa.gz
-rw-rw-r-- 1  52M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz
-rw-rw-r-- 1 6.8M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.Y.fa.gz
-rw-rw-r-- 1  45M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.X.fa.gz
-rw-rw-r-- 1 5.4K Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.MT.fa.gz
-rw-rw-r-- 1  35M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.9.fa.gz
-rw-rw-r-- 1  42M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.8.fa.gz
-rw-rw-r-- 1 3.0G Mar 23 11:21 Homo_sapiens.GRCh38.dna.chromosome.all.fa
-rw-rw-r-- 1  59M Mar 23 11:23 Homo_sapiens.GRCh38.cdna.all.fa.gz
-rw-rw-r-- 1 700M Mar 23 11:31 GRCh38.82.4.bt2
-rw-rw-r-- 1 8.8K Mar 23 11:31 GRCh38.82.3.bt2
-rw-rw-r-- 1 700M Mar 23 11:51 GRCh38.82.2.bt2
-rw-rw-r-- 1 937M Mar 23 11:51 GRCh38.82.1.bt2
-rw-rw-r-- 1  696 Mar 23 12:00 IDX2.stderr
-rw-rw-r-- 1   11 Mar 23 12:10 IDX3.stdout
-rw-rw-r-- 1  975 Mar 23 12:10 IDX3.stderr
-rw-rw-r-- 1 700M Mar 23 12:11 GRCh38.82.rev.2.bt2
-rw-rw-r-- 1 937M Mar 23 12:11 GRCh38.82.rev.1.bt2
drwxrwxr-x 2 4.0K Mar 23 12:14 Homo_sapiens.GRCh38.cdna.all.salmon.idx
-rw-rw-r-- 1 2.2G Mar 23 12:18 Homo_sapiens.GRCh38.cdna.all.kallisto.idx
-rw-rw-r-- 1 4.7K Mar 23 12:48 IDX2.stdout
-rw-rw-r-- 1    0 Mar 23 12:52 IDXHS2.stderr
-rw-rw-r-- 1    7 Mar 23 12:52 IDXHS2.stdout
-rw-rw-r-- 1 700M Mar 23 12:53 GRCh38.82.4.ht2
-rw-rw-r-- 1 8.8K Mar 23 12:53 GRCh38.82.3.ht2
-rw-rw-r-- 1    8 Mar 23 12:53 GRCh38.82.8.ht2
-rw-rw-r-- 1    8 Mar 23 12:53 GRCh38.82.7.ht2
-rw-rw-r-- 1    4 Mar 23 12:53 GRCh38.82.2.ht2
-rw-rw-r-- 1  12K Mar 23 12:55 GRCh38.82.1.ht2
-rw-rw-r-- 1 2.5K Mar 23 12:58 IDXGS.stdout
-rwxr-xr-x 1 1.2K Mar 23 12:58 indexer.sh
drwxrwxr-x 2 4.0K Mar 23 12:58 STAR
-rw-rw-r-- 1 1.1G Mar 23 12:59 GRCh38.82.5.sa
-rw-rw-r-- 1 7.4K Mar 23 12:59 IDXGS.stderr
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ ls -hltrog
total 19G
-rw-rw-r-- 1 1.4G Mar 23 11:16 Homo_sapiens.GRCh38.82.gtf
-rw-rw-r-- 1  38M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.11.fa.gz
-rw-rw-r-- 1  39M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.10.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.18.fa.gz
-rw-rw-r-- 1  23M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.17.fa.gz
-rw-rw-r-- 1  24M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.16.fa.gz
-rw-rw-r-- 1  25M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.15.fa.gz
-rw-rw-r-- 1  26M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.14.fa.gz
-rw-rw-r-- 1  29M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.13.fa.gz
-rw-rw-r-- 1  11M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz
-rw-rw-r-- 1  12M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.21.fa.gz
-rw-rw-r-- 1  18M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.20.fa.gz
-rw-rw-r-- 1  67M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.1.fa.gz
-rw-rw-r-- 1  16M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.19.fa.gz
-rw-rw-r-- 1  55M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.4.fa.gz
-rw-rw-r-- 1  57M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.3.fa.gz
-rw-rw-r-- 1  70M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.2.fa.gz
-rw-rw-r-- 1  46M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz
-rw-rw-r-- 1  50M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.6.fa.gz
-rw-rw-r-- 1  52M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.5.fa.gz
-rw-rw-r-- 1 6.8M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.Y.fa.gz
-rw-rw-r-- 1  45M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.X.fa.gz
-rw-rw-r-- 1 5.4K Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.MT.fa.gz
-rw-rw-r-- 1  35M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.9.fa.gz
-rw-rw-r-- 1  42M Mar 23 11:17 Homo_sapiens.GRCh38.dna.chromosome.8.fa.gz
-rw-rw-r-- 1 3.0G Mar 23 11:21 Homo_sapiens.GRCh38.dna.chromosome.all.fa
-rw-rw-r-- 1  59M Mar 23 11:23 Homo_sapiens.GRCh38.cdna.all.fa.gz
-rw-rw-r-- 1 700M Mar 23 11:31 GRCh38.82.4.bt2
-rw-rw-r-- 1 8.8K Mar 23 11:31 GRCh38.82.3.bt2
-rw-rw-r-- 1 700M Mar 23 11:51 GRCh38.82.2.bt2
-rw-rw-r-- 1 937M Mar 23 11:51 GRCh38.82.1.bt2
-rw-rw-r-- 1  696 Mar 23 12:00 IDX2.stderr
-rw-rw-r-- 1   11 Mar 23 12:10 IDX3.stdout
-rw-rw-r-- 1  975 Mar 23 12:10 IDX3.stderr
-rw-rw-r-- 1 700M Mar 23 12:11 GRCh38.82.rev.2.bt2
-rw-rw-r-- 1 937M Mar 23 12:11 GRCh38.82.rev.1.bt2
drwxrwxr-x 2 4.0K Mar 23 12:14 Homo_sapiens.GRCh38.cdna.all.salmon.idx
-rw-rw-r-- 1 2.2G Mar 23 12:18 Homo_sapiens.GRCh38.cdna.all.kallisto.idx
-rw-rw-r-- 1 4.7K Mar 23 12:48 IDX2.stdout
-rw-rw-r-- 1    0 Mar 23 12:52 IDXHS2.stderr
-rw-rw-r-- 1    7 Mar 23 12:52 IDXHS2.stdout
-rw-rw-r-- 1 700M Mar 23 12:53 GRCh38.82.4.ht2
-rw-rw-r-- 1 8.8K Mar 23 12:53 GRCh38.82.3.ht2
-rw-rw-r-- 1    8 Mar 23 12:53 GRCh38.82.8.ht2
-rw-rw-r-- 1    8 Mar 23 12:53 GRCh38.82.7.ht2
-rw-rw-r-- 1 2.5K Mar 23 12:58 IDXGS.stdout
-rwxr-xr-x 1 1.2K Mar 23 12:58 indexer.sh
-rw-rw-r-- 1 1.1G Mar 23 12:59 GRCh38.82.5.sa
-rw-rw-r-- 1 1.3G Mar 23 13:00 GRCh38.82.2.sa
-rw-rw-r-- 1 1.3G Mar 23 13:00 GRCh38.82.4.sa
drwx------ 2 4.0K Mar 23 13:00 _STARtmp
-rw-rw-r-- 1    0 Mar 23 13:00 IDXSTA.stderr
-rw-rw-r-- 1 1.7G Mar 23 13:01 GRCh38.82.3.sa
drwxrwxr-x 2 4.0K Mar 23 13:01 STAR
-rw-rw-r-- 1  15K Mar 23 13:02 Log.out
-rw-rw-r-- 1  248 Mar 23 13:02 IDXSTA.stdout
-rw-rw-r-- 1 799M Mar 23 13:02 GRCh38.82.7.sa
-rw-rw-r-- 1  13K Mar 23 13:02 IDXGS.stderr
-rw-rw-r-- 1 124M Mar 23 13:02 GRCh38.82.2.ht2
-rw-rw-r-- 1 165M Mar 23 13:02 GRCh38.82.1.ht2
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601770   IDX2       rsh46         Running 6       22:57:28  Thu Mar 23 12:00:40
5601898   IDXHS2     rsh46         Running 6       23:49:29  Thu Mar 23 12:52:41
5601906   IDXGS      rsh46         Running 6       23:53:29  Thu Mar 23 12:56:41
5601911   IDXSTA     rsh46         Running 16      23:57:30  Thu Mar 23 13:00:42

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================

Total Jobs: 4     Active Jobs: 4     Idle Jobs: 0     Blocked Jobs: 0   
[rsh46@login-sand3 GRCh38.82]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601770   IDX2       rsh46         Running 6       22:52:09  Thu Mar 23 12:00:40
5601898   IDXHS2     rsh46         Running 6       23:44:10  Thu Mar 23 12:52:41
5601906   IDXGS      rsh46         Running 6       23:48:10  Thu Mar 23 12:56:41
5601911   IDXSTA     rsh46         Running 16      23:52:11  Thu Mar 23 13:00:42

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================

Total Jobs: 4     Active Jobs: 4     Idle Jobs: 0     Blocked Jobs: 0   
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ 
[rsh46@login-sand3 GRCh38.82]$ cd ~/scratch/Programs/gmap-2017-03-17/
[rsh46@login-sand3 gmap-2017-03-17]$ ls
acinclude.m4  bin        config.log     configure     INSTALL   Makefile.am  NOTICE  src    util
aclocal.m4    ChangeLog  config.site    configure.ac  libtool   Makefile.in  README  tests  VERSION
AUTHORS       config     config.status  COPYING       Makefile  NEWS         share   TODO
[rsh46@login-sand3 gmap-2017-03-17]$ ls share/
GRCh38.82  GRCh38.82.coords  GRCh38.82.sources
[rsh46@login-sand3 gmap-2017-03-17]$ ls
acinclude.m4  bin        config.log     configure     INSTALL   Makefile.am  NOTICE  src    util
aclocal.m4    ChangeLog  config.site    configure.ac  libtool   Makefile.in  README  tests  VERSION
AUTHORS       config     config.status  COPYING       Makefile  NEWS         share   TODO
[rsh46@login-sand3 gmap-2017-03-17]$ pwd
/home/rsh46/scratch/Programs/gmap-2017-03-17
[rsh46@login-sand3 gmap-2017-03-17]$ ls -hltrog
total 2.4M
-rw-r--r-- 1    0 Jun 27  2013 NEWS
-rw-r--r-- 1 1.7K Jun 27  2013 COPYING
-rw-r--r-- 1  181 Jun 27  2013 AUTHORS
-rw-r--r-- 1 4.8K Jul 25  2013 NOTICE
-rw-r--r-- 1   49 Aug 31  2013 TODO
-rw-r--r-- 1 6.0K Dec 12  2014 config.site
-rw-r--r-- 1  252 Aug 11  2015 Makefile.am
-rw-r--r-- 1  796 May 24  2016 acinclude.m4
-rw-r--r-- 1  64K Aug 16  2016 README
-rw-r--r-- 1  21K Dec 16 16:32 configure.ac
-rw-r--r-- 1   10 Mar 18 00:13 VERSION
-rw-r--r-- 1 900K Mar 18 00:21 ChangeLog
-rw-r--r-- 1  43K Mar 18 00:22 aclocal.m4
-rwxr-xr-x 1 640K Mar 18 00:22 configure
-rw-r--r-- 1  27K Mar 18 00:22 Makefile.in
-rw-r--r-- 1  16K Mar 18 00:22 INSTALL
drwxr-xr-x 2 4.0K Mar 18 00:26 config
drwxr-xr-x 4 112K Mar 23 12:53 src
-rw-rw-r-- 1  27K Mar 23 12:53 Makefile
-rwxrwxr-x 1  64K Mar 23 12:53 config.status
drwxr-xr-x 2 4.0K Mar 23 12:53 util
drwxr-xr-x 2 4.0K Mar 23 12:53 tests
-rwxrwxr-x 1 333K Mar 23 12:53 libtool
-rw-rw-r-- 1 112K Mar 23 12:53 config.log
drwxrwxr-x 2 4.0K Mar 23 12:55 bin
drwxrwxr-x 3 4.0K Mar 23 12:57 share
[rsh46@login-sand3 gmap-2017-03-17]$ cd share/
[rsh46@login-sand3 share]$ ls
GRCh38.82  GRCh38.82.coords  GRCh38.82.sources
[rsh46@login-sand3 share]$ ls *
GRCh38.82.coords  GRCh38.82.sources

GRCh38.82:
GRCh38.82.chromosome      GRCh38.82.contig         GRCh38.82.genomecomp           GRCh38.82.ref153offsets64strm
GRCh38.82.chromosome.iit  GRCh38.82.contig.iit     GRCh38.82.maps                 GRCh38.82.ref153positions
GRCh38.82.chrsubset       GRCh38.82.genomebits128  GRCh38.82.ref153offsets64meta  GRCh38.82.version
[rsh46@login-sand3 share]$ ls -hl *
-rw-rw-r-- 1 rsh46 rsh46 1014 Mar 23 12:57 GRCh38.82.coords
-rw-rw-r-- 1 rsh46 rsh46    0 Mar 23 12:56 GRCh38.82.sources

GRCh38.82:
total 6.4G
-rw-rw-r-- 1 rsh46 rsh46  860 Mar 23 12:58 GRCh38.82.chromosome
-rw-rw-r-- 1 rsh46 rsh46 1.5K Mar 23 12:58 GRCh38.82.chromosome.iit
-rw-rw-r-- 1 rsh46 rsh46  184 Mar 23 12:58 GRCh38.82.chrsubset
-rw-rw-r-- 1 rsh46 rsh46 1.3K Mar 23 12:58 GRCh38.82.contig
-rw-rw-r-- 1 rsh46 rsh46 1.7K Mar 23 12:58 GRCh38.82.contig.iit
-rw-rw-r-- 1 rsh46 rsh46 1.1G Mar 23 12:59 GRCh38.82.genomebits128
-rw-rw-r-- 1 rsh46 rsh46 1.1G Mar 23 12:59 GRCh38.82.genomecomp
drwxr-xr-x 2 rsh46 rsh46 4.0K Mar 23 12:56 GRCh38.82.maps
-rw-rw-r-- 1 rsh46 rsh46 129M Mar 23 13:05 GRCh38.82.ref153offsets64meta
-rw-rw-r-- 1 rsh46 rsh46 449M Mar 23 13:05 GRCh38.82.ref153offsets64strm
-rw-rw-r-- 1 rsh46 rsh46 3.7G Mar 23 13:12 GRCh38.82.ref153positions
-rw-rw-r-- 1 rsh46 rsh46   10 Mar 23 12:56 GRCh38.82.version
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ more GRCh38.82.sources
[rsh46@login-sand3 share]$ more GRCh38.82.coords 
# To rename a chromosome, edit each occurrence of that chromosome in the gmap_coordinates
# The strain column has information copied from NCBI md files, but is not otherwise used by gmap_setup
# To exclude a contig, place a '#' sign at the beginning of the line
# The <primary> field means the primary segment for the given (altloc) contig
#contig	gmap_coordinates	linear/circular/<primary>	strain
10	10:1..133797422	linear
11	11:1..135086622	linear
12	12:1..133275309	linear
13	13:1..114364328	linear
14	14:1..107043718	linear
15	15:1..101991189	linear
16	16:1..90338345	linear
17	17:1..83257441	linear
18	18:1..80373285	linear
19	19:1..58617616	linear
1	1:1..248956422	linear
20	20:1..64444167	linear
21	21:1..46709983	linear
22	22:1..50818468	linear
2	2:1..242193529	linear
3	3:1..198295559	linear
4	4:1..190214555	linear
5	5:1..181538259	linear
6	6:1..170805979	linear
7	7:1..159345973	linear
8	8:1..145138636	linear
9	9:1..138394717	linear
MT	MT:1..16569	linear
X	X:1..156040895	linear
Y	Y:1..57227415	linear
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ vi ~/.clusterflow/genomes.config 
[rsh46@login-sand3 share]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601770   IDX2       rsh46         Running 6       22:36:37  Thu Mar 23 12:00:40
5601906   IDXGS      rsh46         Running 6       23:32:38  Thu Mar 23 12:56:41
5601911   IDXSTA     rsh46         Running 16      23:36:39  Thu Mar 23 13:00:42

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================

Total Jobs: 3     Active Jobs: 3     Idle Jobs: 0     Blocked Jobs: 0   
[rsh46@login-sand3 share]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601770   IDX2       rsh46         Running 6       22:34:33  Thu Mar 23 12:00:40
5601906   IDXGS      rsh46         Running 6       23:30:34  Thu Mar 23 12:56:41
5601911   IDXSTA     rsh46         Running 16      23:34:35  Thu Mar 23 13:00:42

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================
5602039   IDXHS2     rsh46         Waiting 6       24:00:00  Thu Mar 23 13:26:03

Total Jobs: 4     Active Jobs: 3     Idle Jobs: 1     Blocked Jobs: 0   
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601770   IDX2       rsh46         Running 6       22:34:22  Thu Mar 23 12:00:40
5601906   IDXGS      rsh46         Running 6       23:30:23  Thu Mar 23 12:56:41
5601911   IDXSTA     rsh46         Running 16      23:34:24  Thu Mar 23 13:00:42

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================
5602039   IDXHS2     rsh46         Waiting 6       24:00:00  Thu Mar 23 13:26:03

Total Jobs: 4     Active Jobs: 3     Idle Jobs: 1     Blocked Jobs: 0   
[rsh46@login-sand3 share]$ ls
GRCh38.82  GRCh38.82.coords  GRCh38.82.sources
[rsh46@login-sand3 share]$ ls -hltrog *
-rw-rw-r-- 1    0 Mar 23 12:56 GRCh38.82.sources
-rw-rw-r-- 1 1014 Mar 23 12:57 GRCh38.82.coords

GRCh38.82:
total 6.4G
-rw-rw-r-- 1   10 Mar 23 12:56 GRCh38.82.version
drwxr-xr-x 2 4.0K Mar 23 12:56 GRCh38.82.maps
-rw-rw-r-- 1 1.7K Mar 23 12:58 GRCh38.82.contig.iit
-rw-rw-r-- 1 1.3K Mar 23 12:58 GRCh38.82.contig
-rw-rw-r-- 1  184 Mar 23 12:58 GRCh38.82.chrsubset
-rw-rw-r-- 1 1.5K Mar 23 12:58 GRCh38.82.chromosome.iit
-rw-rw-r-- 1  860 Mar 23 12:58 GRCh38.82.chromosome
-rw-rw-r-- 1 1.1G Mar 23 12:59 GRCh38.82.genomecomp
-rw-rw-r-- 1 1.1G Mar 23 12:59 GRCh38.82.genomebits128
-rw-rw-r-- 1 449M Mar 23 13:05 GRCh38.82.ref153offsets64strm
-rw-rw-r-- 1 129M Mar 23 13:05 GRCh38.82.ref153offsets64meta
-rw-rw-r-- 1 3.7G Mar 23 13:12 GRCh38.82.ref153positions
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ ls -hltrog */*
-rw-rw-r-- 1   10 Mar 23 12:56 GRCh38.82/GRCh38.82.version
-rw-rw-r-- 1 1.7K Mar 23 12:58 GRCh38.82/GRCh38.82.contig.iit
-rw-rw-r-- 1 1.3K Mar 23 12:58 GRCh38.82/GRCh38.82.contig
-rw-rw-r-- 1  184 Mar 23 12:58 GRCh38.82/GRCh38.82.chrsubset
-rw-rw-r-- 1 1.5K Mar 23 12:58 GRCh38.82/GRCh38.82.chromosome.iit
-rw-rw-r-- 1  860 Mar 23 12:58 GRCh38.82/GRCh38.82.chromosome
-rw-rw-r-- 1 1.1G Mar 23 12:59 GRCh38.82/GRCh38.82.genomecomp
-rw-rw-r-- 1 1.1G Mar 23 12:59 GRCh38.82/GRCh38.82.genomebits128
-rw-rw-r-- 1 449M Mar 23 13:05 GRCh38.82/GRCh38.82.ref153offsets64strm
-rw-rw-r-- 1 129M Mar 23 13:05 GRCh38.82/GRCh38.82.ref153offsets64meta
-rw-rw-r-- 1 3.7G Mar 23 13:12 GRCh38.82/GRCh38.82.ref153positions

GRCh38.82/GRCh38.82.maps:
total 0
[rsh46@login-sand3 share]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601770   IDX2       rsh46         Running 6       22:30:21  Thu Mar 23 12:00:40
5601906   IDXGS      rsh46         Running 6       23:26:22  Thu Mar 23 12:56:41
5601911   IDXSTA     rsh46         Running 16      23:30:23  Thu Mar 23 13:00:42
5602039   IDXHS2     rsh46         Running 16      23:56:24  Thu Mar 23 13:26:43

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================

Total Jobs: 4     Active Jobs: 4     Idle Jobs: 0     Blocked Jobs: 0   
[rsh46@login-sand3 share]$ cd ../
[rsh46@login-sand3 gmap-2017-03-17]$ cd ../erne-2.1.1-linux/
[rsh46@login-sand3 erne-2.1.1-linux]$ ls -hltrog
total 8.0K
drwxrwxr-x 3 4.0K May  3  2016 share
drwxrwxr-x 2 4.0K May  3  2016 bin
[rsh46@login-sand3 erne-2.1.1-linux]$ ls -hltrog share/
total 4.0K
drwxrwxr-x 3 4.0K May  3  2016 doc
[rsh46@login-sand3 erne-2.1.1-linux]$ ls -hltrog share/doc/
total 4.0K
drwxrwxr-x 2 4.0K May  3  2016 erne
[rsh46@login-sand3 erne-2.1.1-linux]$ ls -hltrog share/doc/erne/
total 64K
-rw-r--r-- 1  15K May  3  2016 README
-rw-r--r-- 1  35K May  3  2016 COPYING
-rw-r--r-- 1 6.8K May  3  2016 ChangeLog
-rw-r--r-- 1  263 May  3  2016 AUTHORS
[rsh46@login-sand3 erne-2.1.1-linux]$ ls -hltrog
total 8.0K
drwxrwxr-x 3 4.0K May  3  2016 share
drwxrwxr-x 2 4.0K May  3  2016 bin
[rsh46@login-sand3 erne-2.1.1-linux]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601770   IDX2       rsh46         Running 6       21:58:14  Thu Mar 23 12:00:40
5601906   IDXGS      rsh46         Running 6       22:54:15  Thu Mar 23 12:56:41

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================

Total Jobs: 2     Active Jobs: 2     Idle Jobs: 0     Blocked Jobs: 0   
[rsh46@login-sand3 erne-2.1.1-linux]$ ls
bin  share
[rsh46@login-sand3 erne-2.1.1-linux]$ cd ../
[rsh46@login-sand3 Programs]$ ls
bamtools        cramtools-3.0                 FastQC               infernal-1.1.2               MultiQC             rrna_mt_count              subread-1.5.0-p2-Linux-x86_64
bbmap           cufflinks-2.2.1.Linux_x86_64  fastq_screen_v0.5.2  kallisto_linux-v0.42.5       picard-tools-2.2.4  Salmon-0.8.2_linux_x86_64  svtools
bcftools-1.3.1  deepTools                     gmap-2017-03-17      kallisto_linux-v0.43.0       preseq_v2.0         samtools-1.3.1             telseq
Bismark         Drop-seq_tools-1.12           hisat2-2.0.4         kallisto_linux-v0.43.1       pyBigWig            seqtk                      tophat-2.1.1.Linux_x86_64
bowtie2-2.2.9   dupRadar                      HTSeq-0.6.1p1        local                        pysam               snpEff                     trim_galore_zip
clusterflow     erne-2.1.1-linux              htslib-1.3.1         manta-0.29.6.centos5_x86_64  qualimap_v2.2       STAR
[rsh46@login-sand3 Programs]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601770   IDX2       rsh46         Running 6       21:17:03  Thu Mar 23 12:00:40
5601906   IDXGS      rsh46         Running 6       22:13:04  Thu Mar 23 12:56:41

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================

Total Jobs: 2     Active Jobs: 2     Idle Jobs: 0     Blocked Jobs: 0   
[rsh46@login-sand3 Programs]$ ls -hltrog
total 180K
drwxrwxr-x  2 4.0K Apr 30  2014 cufflinks-2.2.1.Linux_x86_64
drwxrwxr-x  6 4.0K Aug  4  2015 cramtools-3.0
drwxr-xr-x  2 4.0K Oct 29  2015 preseq_v2.0
drwxrwxr-x  4 4.0K Apr  2  2016 tophat-2.1.1.Linux_x86_64
drwxrwxr-x  3 4.0K Apr  4  2016 kallisto_linux-v0.42.5
drwxr-xr-x  6 4.0K Apr 13  2016 manta-0.29.6.centos5_x86_64
drwxr-xr-x  6 4.0K Apr 13  2016 subread-1.5.0-p2-Linux-x86_64
drwxr-xr-x  2 4.0K Apr 27  2016 picard-tools-2.2.4
drwxrwxr-x  4 4.0K May  3  2016 erne-2.1.1-linux
drwxr-xr-x  7 4.0K May  6  2016 STAR
drwxrwxr-x  2 4.0K May  9  2016 trim_galore_zip
drwxrwxr-x  8 4.0K May  9  2016 FastQC
drwxr-xr-x  7  12K May  9  2016 samtools-1.3.1
drwxr-xr-x  8 4.0K May  9  2016 pyBigWig
drwxrwxrwx  5 4.0K May 10  2016 local
drwxrwxr-x  5 4.0K May 18  2016 hisat2-2.0.4
drwxr-xr-x  5 4.0K May 25  2016 snpEff
drwxrwxr-x  3 4.0K Jun  1  2016 svtools
drwxrwxr-x  3 4.0K Jun  2  2016 kallisto_linux-v0.43.0
drwxr-xr-x 13 4.0K Jun  6  2016 deepTools
drwxr-xr-x  5 4.0K Jun  6  2016 htslib-1.3.1
drwxr-xr-x 16 4.0K Jun  6  2016 pysam
drwxr-xr-x  5 4.0K Jun  7  2016 telseq
drwxr-xr-x  9 4.0K Jun  7  2016 bamtools
drwxr-xr-x  6 4.0K Jun  8  2016 bcftools-1.3.1
drwxrwxr-x  5 4.0K Jun  8  2016 bowtie2-2.2.9
drwxrwxr-x  8 4.0K Jun  8  2016 HTSeq-0.6.1p1
drwxr-xr-x  3 4.0K Jun 16  2016 seqtk
drwxr-xr-x  4 4.0K Jul 28  2016 Bismark
drwxr-xr-x 11 4.0K Aug 24  2016 infernal-1.1.2
drwxr-xr-x  8 4.0K Sep  7  2016 clusterflow
drwxr-xr-x  8 4.0K Sep 27 10:54 MultiQC
drwxr-xr-x  7  12K Oct 14 11:08 bbmap
drwxrwxr-x  5 4.0K Nov 18 12:16 qualimap_v2.2
drwxrwxr-x  2 4.0K Nov 21 12:34 dupRadar
drwxr-xr-x  4 4.0K Dec 14 09:58 Drop-seq_tools-1.12
drwxrwxr-x  3 4.0K Feb 27 12:46 rrna_mt_count
drwxr-xr-x  2 4.0K Mar  1 14:36 fastq_screen_v0.5.2
drwxr-xr-x  3 4.0K Mar 22 11:41 kallisto_linux-v0.43.1
drwxr-xr-x  4 4.0K Mar 22 11:41 Salmon-0.8.2_linux_x86_64
drwxr-xr-x  8 4.0K Mar 23 12:56 gmap-2017-03-17
[rsh46@login-sand3 Programs]$ cd gmap-2017-03-17/
[rsh46@login-sand3 gmap-2017-03-17]$ ls
acinclude.m4  AUTHORS  ChangeLog  config.log   config.status  configure.ac  INSTALL  Makefile     Makefile.in  NOTICE  share  tests  util
aclocal.m4    bin      config     config.site  configure      COPYING       libtool  Makefile.am  NEWS         README  src    TODO   VERSION
[rsh46@login-sand3 gmap-2017-03-17]$ cd share
[rsh46@login-sand3 share]$ ls
GRCh38.82  GRCh38.82.coords  GRCh38.82.sources
[rsh46@login-sand3 share]$ ls -hltrog
total 8.0K
-rw-rw-r-- 1    0 Mar 23 12:56 GRCh38.82.sources
-rw-rw-r-- 1 1014 Mar 23 12:57 GRCh38.82.coords
drwxrwxr-x 3 4.0K Mar 23 13:37 GRCh38.82
[rsh46@login-sand3 share]$ ls -hltrog *
-rw-rw-r-- 1    0 Mar 23 12:56 GRCh38.82.sources
-rw-rw-r-- 1 1014 Mar 23 12:57 GRCh38.82.coords

GRCh38.82:
total 18G
-rw-rw-r-- 1   10 Mar 23 12:56 GRCh38.82.version
drwxr-xr-x 2 4.0K Mar 23 12:56 GRCh38.82.maps
-rw-rw-r-- 1 1.7K Mar 23 12:58 GRCh38.82.contig.iit
-rw-rw-r-- 1 1.3K Mar 23 12:58 GRCh38.82.contig
-rw-rw-r-- 1  184 Mar 23 12:58 GRCh38.82.chrsubset
-rw-rw-r-- 1 1.5K Mar 23 12:58 GRCh38.82.chromosome.iit
-rw-rw-r-- 1  860 Mar 23 12:58 GRCh38.82.chromosome
-rw-rw-r-- 1 1.1G Mar 23 12:59 GRCh38.82.genomecomp
-rw-rw-r-- 1 1.1G Mar 23 12:59 GRCh38.82.genomebits128
-rw-rw-r-- 1 449M Mar 23 13:05 GRCh38.82.ref153offsets64strm
-rw-rw-r-- 1 129M Mar 23 13:05 GRCh38.82.ref153offsets64meta
-rw-rw-r-- 1 3.7G Mar 23 13:12 GRCh38.82.ref153positions
-rw-rw-r-- 1  12G Mar 23 13:37 GRCh38.82.sarray
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ 
[rsh46@login-sand3 share]$ cd ~/.clusterflow/pipelines/
[rsh46@login-sand3 pipelines]$ ls
bam_qc_run.config  fastq_bismark.config  fastq_bismark_lux.config  fastq_bowtie2_bgiXten.config  fastq_bowtie.config  fastq_kallisto.config  fastq_tophat_double.config
[rsh46@login-sand3 pipelines]$ 
[rsh46@login-sand3 pipelines]$ 
[rsh46@login-sand3 pipelines]$ 
[rsh46@login-sand3 pipelines]$ cp fastq_kallisto.config fastq_salmon.config
[rsh46@login-sand3 pipelines]$ vi fastq_salmon.config 
[rsh46@login-sand3 pipelines]$ 
[rsh46@login-sand3 pipelines]$ 
[rsh46@login-sand3 pipelines]$ 
[rsh46@login-sand3 pipelines]$ exit
logout
Connection to login.hpc.cam.ac.uk closed.
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ ssh ctr-bfx -l rsh46
Welcome to Ubuntu 16.04.2 LTS (GNU/Linux 4.4.0-64-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

23 packages can be updated.
0 updates are security updates.


*** System restart required ***
Last login: Thu Mar 23 11:35:50 2017 from 172.29.28.97
rsh46@ctr-bfx $ cd /storage/CTR-Projects/CTR_edw23/
rsh46@ctr-bfx $ ls
CTR_edw23_0001
rsh46@ctr-bfx $ cd CTR_edw23_0001/
rsh46@ctr-bfx $ ls
Control-1  MaskedRegion_chr13-mtrr  WholeGenome  WholeGenome_chr13-mtrr-masked  WholeGenome_Complete
rsh46@ctr-bfx $ ls Control-1/
CTR_edw23_0001_Control-1.srtd_mrgd_srtd_dedup_chr13-mtrr-masked.bam
rsh46@ctr-bfx $ exit
logout
Connection to ctr-bfx closed.
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ darwin

        <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
        <>                                                        <>
        <>                        !WARNING!                       <>
        <>                                                        <>
        <>                  HPCs Darwin Facility                  <>
        <>             Unauthorised Access Prohibited             <>
        <>     Use of this system constitutes acceptance of our   <>
        <>         policies - see http://www.hpc.cam.ac.uk/.      <>
        <>                                                        <>
        <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

Last login: Thu Mar 23 11:38:09 2017 from ctr-rsh46.pdn.private.cam.ac.uk
[rsh46@login-sand3 ~]$ cd ~/.clusterflow/modules/
[rsh46@login-sand3 modules]$ perl -C salmon.cfmod.pl 
Error: No run file filename supplied for module salmon
[rsh46@login-sand3 modules]$ perl salmon.cfmod.pl 
Error: No run file filename supplied for module salmon
[rsh46@login-sand3 modules]$ 
[rsh46@login-sand3 modules]$ 
[rsh46@login-sand3 modules]$ 
[rsh46@login-sand3 modules]$ exit
logout
Connection to login.hpc.cam.ac.uk closed.
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ 
rhamilto@ctr $ darwin

        <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
        <>                                                        <>
        <>                        !WARNING!                       <>
        <>                                                        <>
        <>                  HPCs Darwin Facility                  <>
        <>             Unauthorised Access Prohibited             <>
        <>     Use of this system constitutes acceptance of our   <>
        <>         policies - see http://www.hpc.cam.ac.uk/.      <>
        <>                                                        <>
        <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

Last login: Thu Mar 23 16:22:06 2017 from ctr-rsh46.pdn.private.cam.ac.uk
[rsh46@login-sand3 ~]$ cd scratch/CTR-Data/Test/
[rsh46@login-sand3 Test]$ ls
dupRadar.R  runner.sh  Salmon  test.R
[rsh46@login-sand3 Test]$ cd Salmon/
[rsh46@login-sand3 Salmon]$ ls
cf_fastq_bismark_1490284548_submissionlog.txt  test_1_fastqc.html                          test_1.fq.gz_salmon_clusterFlow.txt  test_2_fastqc.html
cf_salmon_1490199742_summary.html              test_1_fastqc.zip                           test_1.fq.gz_salmon_output           test_2_fastqc.zip
cf_salmon_1490199742_summary.txt               test_1.fq.gz                                test_1.fq.gz_salmon.run              test_2.fq.gz
cf_salmon_1490202811_submissionlog.txt         test_1.fq.gz_fastq_bismark_clusterFlow.txt  test_1.fq.gz_trimming_report.txt     test_2.fq.gz_trimming_report.txt
cf_salmon_1490202811_summary.html              test_1.fq.gz_fastq_bismark.run              test_1_trimmed.fq.gz                 test_2_trimmed.fq.gz
cf_salmon_1490202811_summary.txt               test_1.fq.gz_salmon.bam                     test_1_val_1.fq.gz                   test_2_val_2.fq.gz
[rsh46@login-sand3 Salmon]$ myque

SUMMARY OF JOBS FOR USER: <rsh46>

ACTIVE JOBS--------------------
JOBID     JOBNAME    USERNAME      STATE   CORE   REMAINING  STARTTIME
================================================================================
5601770   IDX2       rsh46         Running 6       19:34:47  Thu Mar 23 12:00:40
5601906   IDXGS      rsh46         Running 6       20:30:48  Thu Mar 23 12:56:41
5602818   cf_fastq_b rsh46         Running 3        7:30:21  Thu Mar 23 15:56:14

WAITING JOBS------------------------
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================

BLOCKED JOBS--
JOBID     JOBNAME    USERNAME      STATE   CORE     WCLIMIT  QUEUETIME
================================================================================
5602819   cf_fastq_b rsh46         Waiting 32      24:00:00  Thu Mar 23 15:55:48
5602820   cf_fastq_b rsh46         Waiting 1        2:00:00  Thu Mar 23 15:55:48
5602821   cf_fastq_b rsh46         Waiting 4        6:00:00  Thu Mar 23 15:55:48
5602822   cf_fastq_b rsh46         Waiting 1        0:10:00  Thu Mar 23 15:55:48
5602823   cf_fastq_b rsh46         Waiting 1        0:12:00  Thu Mar 23 15:55:48
5602824   cf_fastq_b rsh46         Waiting 1        0:20:00  Thu Mar 23 15:55:48
5602825   cf_fastq_b rsh46         Waiting 1        0:12:00  Thu Mar 23 15:55:49

Total Jobs: 10    Active Jobs: 3     Idle Jobs: 0     Blocked Jobs: 7   
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ grep salmon ~/bash_eternal_history 
51914 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1051  2017-03-22 15:13:22 cp bt2_indexer.sh salmon_indexer.sh
51914 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1052  2017-03-22 15:13:24 vi salmon_indexer.sh 
51914 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1053  2017-03-22 15:20:50 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1005  2017-03-22 15:20:55 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1005  2017-03-22 15:20:55 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1005  2017-03-22 15:20:55 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1005  2017-03-22 15:20:55 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1008  2017-03-22 15:21:20 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1008  2017-03-22 15:21:20 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1008  2017-03-22 15:21:20 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1008  2017-03-22 15:21:20 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1008  2017-03-22 15:21:20 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1008  2017-03-22 15:21:20 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1008  2017-03-22 15:21:20 salmon
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1010  2017-03-22 15:21:56 sbatch salmon_indexer.sh 
51914 /home/rsh46/.clusterflow/modules  1061  2017-03-22 15:22:50 cp kallisto.cfmod.pl salmon.cfmod.pl
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1011  2017-03-22 15:26:18 salmon version
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1012  2017-03-22 15:26:25 salmon --version
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1018  2017-03-22 15:30:18 ls Homo_sapiens.GRCh38.cdna.all.salmon.idx
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1018  2017-03-22 15:30:18 ls Homo_sapiens.GRCh38.cdna.all.salmon.idx
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1018  2017-03-22 15:30:18 ls Homo_sapiens.GRCh38.cdna.all.salmon.idx
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1018  2017-03-22 15:30:18 ls Homo_sapiens.GRCh38.cdna.all.salmon.idx
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1019  2017-03-22 15:32:57 salmon quant -h
73450 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38  1020  2017-03-22 15:33:14 salmon quant --help-reads
51914 /home/rsh46/.clusterflow/modules  1062  2017-03-22 15:22:53 vi salmon.cfmod.pl 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1053  2017-03-22 15:44:32 cf --genome GRCh38 salmon test_*
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1070  2017-03-22 15:47:17 more test_1.fq.gz_salmon_clusterFlow.txt
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1071  2017-03-22 15:47:38 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1071  2017-03-22 15:47:38 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1071  2017-03-22 15:47:38 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1071  2017-03-22 15:47:38 vi ~/.clusterflow/modules/salmon.cfmod.pl 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1059  2017-03-22 15:50:02 rm cf_salmon_1490197472_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1060  2017-03-22 15:50:10 rm test_1.fq.gz_salmon*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1062  2017-03-22 15:50:19 cf --genome GRCh38 salmon test_*
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1080  2017-03-22 15:54:06 tail test_1.fq.gz_salmon_clusterFlow.txt
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1065  2017-03-22 15:55:32 salmon quant
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1066  2017-03-22 15:55:36 salmon quant -h
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1067  2017-03-22 15:55:41 salmon quant --help-reads
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1081  2017-03-22 15:54:19 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1081  2017-03-22 15:54:19 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1081  2017-03-22 15:54:19 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1081  2017-03-22 15:54:19 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1081  2017-03-22 15:54:19 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1081  2017-03-22 15:54:19 vi ~/.clusterflow/modules/salmon.cfmod.pl 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1071  2017-03-22 15:56:47 rm cf_salmon_1490197820_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1072  2017-03-22 15:57:00 rm -r test_*_salmon*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1075  2017-03-22 15:57:09 cf --genome GRCh38 salmon test_*
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1083  2017-03-22 15:57:17 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1085  2017-03-22 15:58:20 vi ~/.clusterflow/modules/salmon.cfmod.pl 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1086  2017-03-22 16:14:15 rm -r test_*_salmon*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1088  2017-03-22 16:14:18 rm cf_salmon_1490198230_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1088  2017-03-22 16:14:18 rm cf_salmon_1490198230_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1088  2017-03-22 16:14:18 rm cf_salmon_1490198230_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1088  2017-03-22 16:14:18 rm cf_salmon_1490198230_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1088  2017-03-22 16:14:18 rm cf_salmon_1490198230_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1088  2017-03-22 16:14:18 rm cf_salmon_1490198230_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1090  2017-03-22 16:14:28 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1090  2017-03-22 16:14:28 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1090  2017-03-22 16:14:28 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1094  2017-03-22 16:14:56 rm cf_salmon_1490199268_submissionlog.txt test_1.fq.gz_salmon.run 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1099  2017-03-22 16:18:13 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1101  2017-03-22 16:18:35 rm cf_salmon_1490199268_submissionlog.txt test_1.fq.gz_salmon.run 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1102  2017-03-22 16:18:38 rm cf_salmon_1490199494_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1104  2017-03-22 16:19:10 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1104  2017-03-22 16:19:10 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1104  2017-03-22 16:19:10 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1106  2017-03-22 16:19:45 rm cf_salmon_1490199268_submissionlog.txt test_1.fq.gz_salmon.run 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1107  2017-03-22 16:19:47 rm cf_salmon_1490199550_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1107  2017-03-22 16:19:47 rm cf_salmon_1490199550_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1107  2017-03-22 16:19:47 rm cf_salmon_1490199550_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1107  2017-03-22 16:19:47 rm cf_salmon_1490199550_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1107  2017-03-22 16:19:47 rm cf_salmon_1490199550_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1107  2017-03-22 16:19:47 rm cf_salmon_1490199550_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1112  2017-03-22 16:22:13 rm test_1.fq.gz_salmon_clusterFlow.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1115  2017-03-22 16:22:21 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1115  2017-03-22 16:22:21 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1115  2017-03-22 16:22:21 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1115  2017-03-22 16:22:21 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1115  2017-03-22 16:22:21 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1115  2017-03-22 16:22:21 cf --genome GRCh38 salmon test_*
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1089  2017-03-22 16:02:31 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1089  2017-03-22 16:02:31 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1089  2017-03-22 16:02:31 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1089  2017-03-22 16:02:31 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1089  2017-03-22 16:02:31 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1089  2017-03-22 16:02:31 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1089  2017-03-22 16:02:31 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1091  2017-03-22 16:30:58 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1093  2017-03-22 16:32:33 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1093  2017-03-22 16:32:33 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1093  2017-03-22 16:32:33 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1093  2017-03-22 16:32:33 vi ~/.clusterflow/modules/salmon.cfmod.pl 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2  1126  2017-03-22 16:33:56 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2  1126  2017-03-22 16:33:56 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2  1126  2017-03-22 16:33:56 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2  1126  2017-03-22 16:33:56 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2  1126  2017-03-22 16:33:56 cf --genome GRCh38 salmon test_*
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1095  2017-03-22 16:34:38 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1095  2017-03-22 16:34:38 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1095  2017-03-22 16:34:38 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1095  2017-03-22 16:34:38 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1095  2017-03-22 16:34:38 vi ~/.clusterflow/modules/salmon.cfmod.pl 
51914 /home/rsh46/scratch/CTR-Data/Test/Salmon  1101  2017-03-22 16:41:38 tail test_1.fq.gz_salmon.sam
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2  1144  2017-03-22 16:45:48 samtools view test_1.fq.gz_salmon.sam | more
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2  1154  2017-03-22 17:04:11 samtools view test_1.fq.gz_salmon.sam | more
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1156  2017-03-22 17:04:20 cd test_1.fq.gz_salmon_output
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1157  2017-03-22 17:04:21 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1157  2017-03-22 17:04:21 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1157  2017-03-22 17:04:21 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1157  2017-03-22 17:04:21 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1157  2017-03-22 17:04:21 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1158  2017-03-22 17:04:28 more quant.sf 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1159  2017-03-22 17:04:37 more quant.genes.sf 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1160  2017-03-22 17:04:40 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1160  2017-03-22 17:04:40 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1160  2017-03-22 17:04:40 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1160  2017-03-22 17:04:40 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1160  2017-03-22 17:04:40 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2/test_1.fq.gz_salmon_output  1160  2017-03-22 17:04:40 ls
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon/T2  1167  2017-03-22 17:05:03 mv test_1.fq.gz_salmon.sam test_1.fq.gz_salmon.bam
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1171  2017-03-22 17:12:58 rm cf_salmon_1490199742_submissionlog.txt 
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1172  2017-03-22 17:13:03 rm *salmon*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1174  2017-03-22 17:13:11 rm -r *salmon*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1177  2017-03-22 17:13:31 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1177  2017-03-22 17:13:31 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1177  2017-03-22 17:13:31 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1177  2017-03-22 17:13:31 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1177  2017-03-22 17:13:31 cf --genome GRCh38 salmon test_*
73450 /home/rsh46/scratch/CTR-Data/Test/Salmon  1177  2017-03-22 17:13:31 cf --genome GRCh38 salmon test_*
8757 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38.82  1025  2017-03-23 11:18:45 mv salmon_indexer.sh indexer.sh
8757 /home/rsh46/.clusterflow/modules  1167  2017-03-23 14:45:49 vi salmon.cfmod.pl 
8757 /home/rsh46/.clusterflow/modules  1175  2017-03-23 14:51:41 vi salmon.cfmod.pl 
8757 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38.82  1181  2017-03-23 15:07:57 salmon --version
8757 /home/rsh46/scratch/Genomes/Homo_sapiens/GRCh38.82  1182  2017-03-23 15:09:43 vi ~/.clusterflow/modules/salmon.cfmod.pl 
17280 /home/rsh46/.clusterflow/pipelines  1117  2017-03-23 15:18:56 cp fastq_kallisto.config fastq_salmon.config
17280 /home/rsh46/.clusterflow/pipelines  1118  2017-03-23 15:19:00 vi fastq_salmon.config 
17280 /home/rsh46/.clusterflow/pipelines  1118  2017-03-23 15:19:00 vi fastq_salmon.config 
17280 /home/rsh46/.clusterflow/pipelines  1118  2017-03-23 15:19:00 vi fastq_salmon.config 
17280 /home/rsh46/.clusterflow/pipelines  1118  2017-03-23 15:19:00 vi fastq_salmon.config 
16868 /home/rsh46/.clusterflow/modules  1002  2017-03-23 16:22:25 perl -C salmon.cfmod.pl 
16868 /home/rsh46/.clusterflow/modules  1003  2017-03-23 16:22:34 perl salmon.cfmod.pl 
16868 /home/rsh46/.clusterflow/modules  1003  2017-03-23 16:22:34 perl salmon.cfmod.pl 
16868 /home/rsh46/.clusterflow/modules  1003  2017-03-23 16:22:34 perl salmon.cfmod.pl 
16868 /home/rsh46/.clusterflow/modules  1003  2017-03-23 16:22:34 perl salmon.cfmod.pl 
[rsh46@login-sand3 Salmon]$ ls
cf_fastq_bismark_1490284548_submissionlog.txt  test_1_fastqc.html                          test_1.fq.gz_salmon_clusterFlow.txt  test_2_fastqc.html
cf_salmon_1490199742_summary.html              test_1_fastqc.zip                           test_1.fq.gz_salmon_output           test_2_fastqc.zip
cf_salmon_1490199742_summary.txt               test_1.fq.gz                                test_1.fq.gz_salmon.run              test_2.fq.gz
cf_salmon_1490202811_submissionlog.txt         test_1.fq.gz_fastq_bismark_clusterFlow.txt  test_1.fq.gz_trimming_report.txt     test_2.fq.gz_trimming_report.txt
cf_salmon_1490202811_summary.html              test_1.fq.gz_fastq_bismark.run              test_1_trimmed.fq.gz                 test_2_trimmed.fq.gz
cf_salmon_1490202811_summary.txt               test_1.fq.gz_salmon.bam                     test_1_val_1.fq.gz                   test_2_val_2.fq.gz
[rsh46@login-sand3 Salmon]$ cf --genome GRCh38 salmon test_[12].fq.gz
==================================================
 Cluster Flow v0.5 dev
 Launching pipeline salmon_1490286400
==================================================

Filenames look like paired-end input. Setting --paired
Specify --single to prevent this behaviour.

Processing 2 files in 1 run.

Pipeline to be used:

#salmon


Processing files (one dot per file):
..
Finished processing files.

Jobs submitted.

[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ ls
cf_fastq_bismark_1490284548_submissionlog.txt  test_1_fastqc.html                          test_1.fq.gz_salmon_output        test_2.fq.gz
cf_salmon_1490199742_summary.html              test_1_fastqc.zip                           test_1.fq.gz_salmon.run           test_2.fq.gz_trimming_report.txt
cf_salmon_1490199742_summary.txt               test_1.fq.gz                                test_1.fq.gz_trimming_report.txt  test_2_trimmed.fq.gz
cf_salmon_1490202811_submissionlog.txt         test_1.fq.gz_fastq_bismark_clusterFlow.txt  test_1_trimmed.fq.gz              test_2_val_2.fq.gz
cf_salmon_1490202811_summary.html              test_1.fq.gz_fastq_bismark.run              test_1_val_1.fq.gz
cf_salmon_1490202811_summary.txt               test_1.fq.gz_salmon.bam                     test_2_fastqc.html
cf_salmon_1490286400_submissionlog.txt         test_1.fq.gz_salmon_clusterFlow.txt         test_2_fastqc.zip
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ 
[rsh46@login-sand3 Salmon]$ cd ~/.clusterflow/modules/
[rsh46@login-sand3 modules]$ vi salmon.cfmod.pl 

#!/usr/bin/env perl
use warnings;
use strict;
use Getopt::Long;
use FindBin qw($Bin);
use lib "$FindBin::Bin/../source";
use CF::Constants;
use CF::Helpers;

##########################################################################
# Copyright 2014, Philip Ewels (phil.ewels@scilifelab.se)                #
# Copyright 2017, Russell Hamilton (rsh46@cam.ac.uk)                     #
#   (module adapted from kallisto.cfmod.pl                               #
# This file is part of Cluster Flow.                                     #
#                                                                        #
# Cluster Flow is free software: you can redistribute it and/or modify   #
# it under the terms of the GNU General Public License as published by   #
# the Free Software Foundation, either version 3 of the License, or      #
# (at your option) any later version.                                    #
#                                                                        #
# Cluster Flow is distributed in the hope that it will be useful,        #
# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
# GNU General Public License for more details.                           #
#                                                                        #
# You should have received a copy of the GNU General Public License      #
# along with Cluster Flow.  If not, see <http://www.gnu.org/licenses/>.  #
##########################################################################

# Module requirements
my %requirements = (
        'cores'         => '8',
        'memory'        => '15G',
        'modules'       => ['salmon', 'samtools'],
        'references'=> 'salmon',
        'time'          => sub {
                my $cf = $_[0];
                my $num_files = $cf->{'num_starting_merged_aligned_files'};
                $num_files = ($num_files > 0) ? $num_files : 1;
                return CF::Helpers::minutes_to_timestamp ($num_files * 2 * 60);
        }
);

# Help text
my $helptext = "".("-"x17)."\n Salmon\n".("-"x17)."\n
Salmon is a tool for quantifying the expression of transcripts using 
RNA-seq data. Salmon uses new algorithms (specifically, coupling the 
concept of quasi-mapping with a two-phase inference procedure) to 
provide accurate expression estimates very quickly (i.e. wicked-fast) 
and while using little memory. Salmon performs its inference using an 
expressive and realistic model of RNA-seq data that takes into 
account experimental attributes and biases commonly observed in real 
RNA-seq data.\n\n";

# Setup
my %cf = CF::Helpers::module_start(\%requirements, $helptext);

# MODULE
# Check that we have a genome defined
if(!defined($cf{'refs'}{'salmon'})){
   die "\n\n###CF Error: No salmon transcriptome index found in run file $cf{run_fn} for job $cf{job_id}. Exiting.. ###";
} else {
    warn "\nPseudoaligning against $cf{refs}{salmon}\n\n";
}

open (RUN,'>>',$cf{'run_fn'}) or die "###CF Error: Can't write to $cf{run_fn}: $!";

# Print version information about the module.
my $version = `salmon --version`;
warn "---------- Salmon version information ----------\n";
warn $version;
warn "\n------- End of Salmon version information ------\n";
if($version =~ /version : ([\d\.]+)/){
  warn "###CFVERS salmon\t$1\n\n";
  }

# Separate file names into single end and paired end
my ($se_files, $pe_files) = CF::Helpers::is_paired_end(\%cf, @{$cf{'prev_job_files'}});

# Go through each single end files and run salmon
if($se_files && scalar(@$se_files) > 0){
        foreach my $file (@$se_files){
                my $timestart = time;

                my $output_dir = $file."_salmon_output";
                my $output_fn  = $file."_salmon.bam";

                my $command = "salmon quant --libType A -p $cf{cores} -i $cf{refs}{salmon} -o $output_dir -r $file --geneMap $cf{refs}{gtf} --writeMappings | samtools view -Sb - > $output_fn";

                warn "\n###CFCMD $command\n\n";

                if(!system ($command)){
                        # salmon worked - print out resulting filenames
                        my $duration =  CF::Helpers::parse_seconds(time - $timestart);
                        warn "###CF salmon (SE mode) successfully exited, took $duration..\n";
                        if(-e $output_fn){
                                print RUN "$cf{job_id}\t$output_fn\n";
                        } else {
                                warn "\n###CF Error! salmon output file $output_fn not found..\n";
                        }
                } else {
                        warn "\n###CF Error! salmon (SE mode) failed exited in an error state for input file '$file': $? $!\n\n";
                }
        }
}

# Go through the paired end files and run salmon
if($pe_files && scalar(@$pe_files) > 0){
        foreach my $files_ref (@$pe_files){
                my @files = @$files_ref;
                if(scalar(@files) == 2){
                        my $timestart = time;

                        my $output_dir = $files[0]."_salmon_output";
                        my $output_fn  = $files[0]."_salmon.bam";

                        my $command = "salmon quant --libType A -p $cf{cores} -i $cf{refs}{salmon} -o $output_dir -1 $files[0] -2 $files[1] --geneMap $cf{refs}{gtf} --writeMappings | samtools view -Sb - > $output_fn";
                        warn "\n###CFCMD $command\n\n";

                        if(!system ($command)){
                                # salmon worked - print out resulting filenames
                                my $duration =  CF::Helpers::parse_seconds(time - $timestart);
                                warn "###CF salmon (PE mode) successfully exited, took $duration..\n";
                                if(-e $output_fn){
                                        print RUN "$cf{job_id}\t$output_fn\n";
                                } else {
                                        warn "\n###CF Error! salmon output file $output_fn not found..\n";
                                }
                        } else {
                                warn "\n###CF Error! salmon (PE mode) exited in an error state for input file '".$files[0]."': $? $!\n\n";
                        }

                } else {
                        warn "\n###CF Error! salmon paired end files had ".scalar(@files)." input files instead of 2\n";
                }
        }
}

close (RUN);
